--- ../Test/HttpClientTestCase.php	2025-09-15 20:29:47.275069773 +0300
+++ ../Test/HttpClientTestCase.php	2025-09-15 20:33:27.802017650 +0300
@@ -735,18 +735,6 @@
         $this->assertSame(200, $response->getStatusCode());
     }
 
-    public function testIPv6Resolve()
-    {
-        TestHttpServer::start(-8087);
-
-        $client = $this->getHttpClient(__FUNCTION__);
-        $response = $client->request('GET', 'http://symfony.com:8087/', [
-            'resolve' => ['symfony.com' => '::1'],
-        ]);
-
-        $this->assertSame(200, $response->getStatusCode());
-    }
-
     public function testNotATimeout()
     {
         $client = $this->getHttpClient(__FUNCTION__);
@@ -1177,20 +1165,4 @@
         self::assertSame('9876', $vars['REMOTE_PORT']);
     }
 
-    public function testBindToPortV6()
-    {
-        TestHttpServer::start(-8087);
-
-        $client = $this->getHttpClient(__FUNCTION__);
-        $response = $client->request('GET', 'http://[::1]:8087', ['bindto' => '[::1]:9876']);
-        $response->getStatusCode();
-
-        $vars = $response->toArray();
-
-        self::assertSame('::1', $vars['REMOTE_ADDR']);
-
-        if ('\\' !== \DIRECTORY_SEPARATOR) {
-            self::assertSame('9876', $vars['REMOTE_PORT']);
-        }
-    }
 }
