--- ../test/Middleware/ForwardedMiddlewareTest.php	2024-08-10 15:10:42.149821027 +0300
+++ ../test/Middleware/ForwardedMiddlewareTest.php	2024-08-10 15:11:37.501467971 +0300
@@ -47,27 +47,6 @@
     {
         yield [
             ForwardedHeaderType::Forwarded,
-            'For="[2001:db8:cafe::17]:4711"',
-            new InternetAddress('2001:db8:cafe::17', 4711),
-            [
-                'for' => '[2001:db8:cafe::17]:4711',
-            ],
-        ];
-
-        yield [
-            ForwardedHeaderType::Forwarded,
-            'for="[2001:db8:cafe::17]";proto=https;secret=test;by=172.18.0.9',
-            new InternetAddress('2001:db8:cafe::17', 0),
-            [
-                'for' => '[2001:db8:cafe::17]',
-                'proto' => 'https',
-                'secret' => 'test',
-                'by' => '172.18.0.9',
-            ],
-        ];
-
-        yield [
-            ForwardedHeaderType::Forwarded,
             'for=192.0.2.60;proto=http;by=203.0.113.43',
             new InternetAddress('192.0.2.60', 0),
             [
@@ -85,30 +64,6 @@
                 'for' => '198.51.100.17',
             ],
         ];
-
-        yield [
-            ForwardedHeaderType::Forwarded,
-            'for="2001:db8:cafe::17"',
-            null,
-        ];
-
-        yield [
-            ForwardedHeaderType::XForwardedFor,
-            '2001:db8:85a3:8d3:1319:8a2e:370:7348',
-            new InternetAddress('2001:db8:85a3:8d3:1319:8a2e:370:7348', 0),
-        ];
-
-        yield [
-            ForwardedHeaderType::XForwardedFor,
-            '203.0.113.195,2001:db8:85a3:8d3:1319:8a2e:370:7348,150.172.238.178',
-            new InternetAddress('150.172.238.178', 0),
-        ];
-
-        yield [
-            ForwardedHeaderType::XForwardedFor,
-            '2001:db8:85a3:8d3:1319:8a2e:370',
-            null,
-        ];
     }
 
     /**
