--- a/configure.ac	2017-07-27 20:08:41.000000000 +0300
+++ b/configure.ac	2017-07-27 20:10:46.000000000 +0300
@@ -12,7 +12,14 @@
 AC_CHECK_OCAML_BINDING([bytes],[],[],[1])
 INC="$INC `$OCAMLFIND query bytes`"
 
-AC_CHECK_OCAML_BINDING([ssl])
+AC_ARG_ENABLE([ssl],
+    AS_HELP_STRING([--disable-ssl],[don't use ssl]))
+if test "x$enable_ssl" = "xno" ; then
+    W_SSL=""
+    w_SSL="no"
+else
+    AC_CHECK_OCAML_BINDING([ssl])
+fi
 
 if test "$w_SSL" == "yes"; then
   INC="$INC `$OCAMLFIND query ssl`"
