--- ../CMakeLists.txt	2025-03-24 09:34:28.699661066 +0200
+++ ../CMakeLists.txt	2025-03-24 09:34:33.364780629 +0200
@@ -439,10 +439,6 @@
       RENAME ${APP_ID}.png)
   endforeach()
 
-  install(
-    FILES ${PROJECT_SOURCE_DIR}/LICENSE
-    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/licenses/${PROJECT_NAME_LOWERCASE})
-
   # On Debian-based distros, compress the changelog file and install the copyright file.
   # FIXME: Find out if it's possible to only do this when creating a deb package with CPack.
   if(EXISTS /etc/debian_version)
