--- ../test/AsyncTestCaseTest.php	2026-07-25 11:29:55.113162305 +0300
+++ ../test/AsyncTestCaseTest.php	2026-07-25 12:15:22.366729407 +0300
@@ -5,6 +5,8 @@
 use Amp\DeferredFuture;
 use Amp\Future;
 use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\Depends;
 use Revolt\EventLoop;
 use function Amp\async;
 use function Amp\delay;
@@ -80,16 +82,14 @@
         });
     }
 
-    public function provideArguments(): array
+    public static function provideArguments(): array
     {
         return [
             ['foo', 42, true],
         ];
     }
 
-    /**
-     * @dataProvider provideArguments
-     */
+    #[DataProvider('provideArguments')]
     public function testArgumentSupport(string $foo, int $bar, bool $baz): void
     {
         self::assertSame('foo', $foo);
@@ -97,9 +97,7 @@
         self::assertTrue($baz);
     }
 
-    /**
-     * @depends testReturningFuture
-     */
+    #[Depends('testReturningFuture')]
     public function testReturnValueFromDependentTest(string $value = null): void
     {
         self::assertSame('value', $value);
