diff --git a/src/AsyncTestCase.php b/src/AsyncTestCase.php
index 9bdcf93..2eb0dde 100644
--- a/src/AsyncTestCase.php
+++ b/src/AsyncTestCase.php
@@ -27,15 +27,6 @@ abstract class AsyncTestCase extends PHPUnitTestCase
 
     private bool $setUpInvoked = false;
 
-    /**
-     * @codeCoverageIgnore Invoked before code coverage data is being collected.
-     */
-    final public function setName(string $name): void
-    {
-        parent::setName($name);
-        $this->realTestName = $name;
-    }
-
     protected function setUp(): void
     {
         $this->setUpInvoked = true;
@@ -114,6 +105,7 @@ final protected function runAsyncTest(mixed ...$args): mixed
 
     final protected function runTest(): mixed
     {
+        $this->realTestName = $this->name();
         parent::setName('runAsyncTest');
         return parent::runTest();
     }
