--- ../src/Makefile.am	2025-11-11 10:45:53.510794043 +0200
+++ ../src/Makefile.am	2025-11-11 10:46:13.196809695 +0200
@@ -3,6 +3,6 @@
 parsecfg.c parsecfg.h proto.h shhopt.c shhopt.h strlib.c strlib.h variable.h \
 ftpmain.c message.c cache.c sub.c filedata.c log.c
 
-LDADD=$(INTLLIBS)
+LDADD=$(LIBINTL)
 AM_CPPFLAGS=-DLOCALEDIR=\""$(localedir)"\"
 INCLUDES=-I$(top_srcdir)/intl -I$(top_builddir)/intl
