--- ../tests/UtilsTest.php.org	2026-07-09 18:44:35.657657197 +0300
+++ ../tests/UtilsTest.php	2026-07-09 18:45:42.033556300 +0300
@@ -13,6 +13,7 @@
 use GuzzleHttp\Promise\RejectionException;
 use GuzzleHttp\Promise\TaskQueue;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 class UtilsTest extends TestCase
 {
@@ -328,9 +329,7 @@
         $this->assertSame('ab', $result);
     }
 
-    /**
-     * @dataProvider rejectsParentExceptionProvider
-     */
+    #[DataProvider('rejectsParentExceptionProvider')]
     public function testRejectsParentExceptionWhenException(PromiseInterface $promise): void
     {
         $promise->then(
