--- ../CMakeLists.txt	2025-07-02 14:32:05.773461887 +0300
+++ ../CMakeLists.txt	2025-07-02 14:32:45.857502918 +0300
@@ -177,7 +177,7 @@
 endif()
 
 if(NOT APPLE)
-    set(CMAKE_SHARED_LINKER_FLAGS -Wl,-z,noexecstack,-z,relro,-z,now)
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,noexecstack,-z,relro,-z,now")
 endif()
 
 # Whether to fail the build when compiling s2n's portable C code with non-portable assembly optimizations. Doing this
