--- a/configure.ac	2017-07-29 13:57:01.000000000 +0300
+++ b/configure.ac	2017-07-29 13:59:39.000000000 +0300
@@ -19,6 +19,9 @@
 AC_C_BIGENDIAN(AC_DEFINE([BIGENDIAN], [1], [The target is big endian]),[])
 
 # Ogg variables
+AC_ARG_ENABLE([ogg],
+    AS_HELP_STRING([--disable-ogg],[don't use ogg]))
+if test "x$enable_ogg" != "xno" ; then
 OCAMLOGG_REQUIRES="ogg"
 OCAMLOGG_META="package \"ogg\" (
   requires = \"flac ogg\"
@@ -30,6 +33,7 @@
 OCAMLOGG_FILES="ogg_flac_stubs.c ogg_flac.ml ogg_flac.mli ogg_demuxer_flac_decoder.mli ogg_demuxer_flac_decoder.ml"
 OCAMLOGG_LIBS="ogg"
 OCAMLOGG_INC=
+fi
 
 AC_ARG_WITH([ogg-dir],AS_HELP_STRING([--with-ogg-dir=path],[use "path" as the location of ocaml-ogg (autodetected by default)]))
 if test -z "$with_ogg_dir"; then
