--- ../configure.ac	2019-08-11 18:55:52.217069153 +0300
+++ ../configure.ac	2019-08-11 18:56:41.667183333 +0300
@@ -34,6 +34,9 @@
 fi
 
 # Ogg variables
+AC_ARG_ENABLE([ogg],
+    AS_HELP_STRING([--disable-ogg],[don't use ogg]))
+if test "x$enable_ogg" != "xno" ; then
 OCAMLOGG_REQUIRES="ogg"
 OCAMLOGG_META="package \"ogg\" (
   requires = \"flac ogg\"
@@ -64,6 +67,7 @@
   || with_ogg_dir=$PWD/$with_ogg_dir
   OCAMLOGG_INC="$with_ogg_dir"
 fi
+fi
 AC_SUBST(OCAMLOGG_INC)
 AC_SUBST(OCAMLOGG_FILES)
 AC_SUBST(OCAMLOGG_LIBS)
