--- a/GNUmakefile	2017-01-26 02:03:35.273657660 +0200
+++ b/GNUmakefile	2017-01-26 02:04:34.596654989 +0200
@@ -50,13 +50,7 @@
 # "OBJS += sandbox-null.o" and recompile.
 # PLEASE TELL ME IF IT DOESN'T WORK.
 
-ifeq ($(shell pkg-config --exists libseccomp && echo 1),1)
-OBJS	+= sandbox-seccomp.o
-LIBADD	+= $(shell pkg-config --libs libseccomp)
-CFLAGS	+= $(shell pkg-config --cflags libseccomp)
-else
 OBJS	+= sandbox-null.o
-endif
 
 else ifeq ($(shell uname), Darwin)
 
