--- a/GNUmakefile	2017-01-26 01:56:05.239424211 +0200
+++ b/GNUmakefile	2017-01-26 01:56:38.786422701 +0200
@@ -2,9 +2,9 @@
 # not overly system-specific, please submit a bug report.
 
 PREFIX	?= /usr/local
-MAN1DIR	?= $(PREFIX)/man/man1
+MAN1DIR	?= $(PREFIX)/share/man/man1
 BINDIR	?= $(PREFIX)/bin
-CFLAGS	+= -g -W -Wall -DHAVE_CONFIG_H
+CFLAGS	+= -W -Wall -DHAVE_CONFIG_H
 OBJS 	 = acctproc.o \
 	   base64.o \
 	   certproc.o \
@@ -42,8 +42,6 @@
 CFLAGS	+= -DMUSL_LIBC
 endif
 
-CFLAGS	+= -I/usr/local/include/libressl
-LDFLAGS += -L/usr/local/lib
 OBJS	+= util-portable.o
 
 # Do we have libseccomp installed?
