--- a/cmake/modules/FindPythonQt.cmake	2016-08-16 21:39:18.000000000 +0300
+++ b/cmake/modules/FindPythonQt.cmake	2017-12-30 16:31:34.843909590 +0200
@@ -3,9 +3,11 @@
 # Sets PYTHONQT_FOUND, PYTHONQT_INCLUDE_DIR, PYTHONQT_LIBRARY, PYTHONQT_LIBRARIES
 #
 
-find_path(PYTHONQT_INSTALL_DIR include/PythonQt/PythonQt.h DOC "Directory where PythonQt was installed.")
-find_path(PYTHONQT_INCLUDE_DIR PythonQt.h "${PYTHONQT_INSTALL_DIR}/include/PythonQt" DOC "Path to the PythonQt include directory")
-find_library(PYTHONQT_LIBRARY NAMES PythonQt QtPython PATHS "${PYTHONQT_INSTALL_DIR}/lib" DOC "The PythonQt library.")
+find_path(PYTHONQT_INSTALL_DIR NAMES include/PythonQt/PythonQt.h src/PythonQt.h DOC "Directory where PythonQt was installed.")
+find_path(PYTHONQT_INCLUDE_DIR PythonQt.h PATHS "${PYTHONQT_INSTALL_DIR}/include/PythonQt" "${PYTHONQT_INSTALL_DIR}/src" DOC "Path to the PythonQt include directory")
+file(GLOB PYTHONQT_LIB_FILE LIST_DIRECTORIES false RELATIVE "${PYTHONQT_INSTALL_DIR}/lib" "${PYTHONQT_INSTALL_DIR}/lib/*PythonQt-Qt[4-9]*.so")
+string(REGEX REPLACE "^lib(.+)\\.so$" "\\1" PYTHONQT_LIB "${PYTHONQT_LIB_FILE}")
+find_library(PYTHONQT_LIBRARY NAMES PythonQt QtPython "${PYTHONQT_LIB}" PATHS "${PYTHONQT_INSTALL_DIR}/lib" DOC "The PythonQt library.")
 
 mark_as_advanced(PYTHONQT_INSTALL_DIR)
 mark_as_advanced(PYTHONQT_INCLUDE_DIR)
--- a/cmake/modules/FindPythonQt_QtAll.cmake	2016-08-16 21:39:18.000000000 +0300
+++ b/cmake/modules/FindPythonQt_QtAll.cmake	2017-12-30 16:46:46.521451083 +0200
@@ -4,9 +4,12 @@
 #
 
 
-find_path(PYTHONQT_QTALL_INSTALL_DIR NAMES include/PythonQt/PythonQt_QtAll.h include/PythonQt/extensions/PythonQt_QtAll/PythonQt_QtAll.h DOC "Directory where PythonQt_QtAll.h was installed.")
-find_path(PYTHONQT_QTALL_INCLUDE_DIR PythonQt_QtAll.h PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt" "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt/extensions/PythonQt_QtAll" ${CMAKE_INCLUDE_PATH} ${CMAKE_INSTALL_PREFIX}/include ${CMAKE_INSTALL_PREFIX}/include/PythonQt ${CMAKE_INSTALL_PREFIX}/include/PythonQt/extensions/PythonQt_QtAll DOC "Path to the PythonQt_QtAll include directory")
-find_library(PYTHONQT_QTALL_LIBRARY NAMES PythonQt_QtAll QtPython_QtAll PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" DOC "The PythonQt_QtAll library.")
+find_path(PYTHONQT_QTALL_INSTALL_DIR NAMES include/PythonQt/PythonQt_QtAll.h include/PythonQt/extensions/PythonQt_QtAll/PythonQt_QtAll.h extensions/PythonQt_QtAll/PythonQt_QtAll.h DOC "Directory where PythonQt_QtAll.h was installed.")
+find_path(PYTHONQT_QTALL_INCLUDE_DIR PythonQt_QtAll.h PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt" "${PYTHONQT_QTALL_INSTALL_DIR}/include/PythonQt/extensions/PythonQt_QtAll" "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" ${CMAKE_INCLUDE_PATH} ${CMAKE_INSTALL_PREFIX}/include ${CMAKE_INSTALL_PREFIX}/include/PythonQt ${CMAKE_INSTALL_PREFIX}/include/PythonQt/extensions/PythonQt_QtAll DOC "Path to the PythonQt_QtAll include directory")
+file(GLOB PYTHONQT_QTALL_LIB_FILE LIST_DIRECTORIES false RELATIVE "${PYTHONQT_QTALL_INSTALL_DIR}/lib" "${PYTHONQT_QTALL_INSTALL_DIR}/lib/*PythonQt_QtAll-Qt[4-9]*.so")
+string(REGEX REPLACE "^lib(.+)\\.so$" "\\1" PYTHONQT_QTALL_LIB "${PYTHONQT_QTALL_LIB_FILE}")
+
+find_library(PYTHONQT_QTALL_LIBRARY NAMES PythonQt_QtAll QtPython_QtAll "${PYTHONQT_QTALL_LIB}" PATHS "${PYTHONQT_QTALL_INSTALL_DIR}/extensions/PythonQt_QtAll" "${PYTHONQT_QTALL_INSTALL_DIR}/lib" DOC "The PythonQt_QtAll library.")
 
 mark_as_advanced(PYTHONQT_QTALL_INSTALL_DIR)
 mark_as_advanced(PYTHONQT_QTALL_INCLUDE_DIR)
