--- ../src/pingu.c	2024-08-20 16:22:22.536218773 +0000
+++ ../src/pingu.c	2024-08-20 16:24:53.400500545 +0000
@@ -10,7 +10,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-
+#include <libgen.h>
 #include <ev.h>
 
 #include "log.h"
@@ -39,6 +39,8 @@
 int pingu_verbose = 0, pid_file_fd = 0, pingu_daemonize = 0;
 char *pid_file = DEFAULT_PIDFILE;
 
+char *basename (char *fname);
+
 static void print_version(const char *program)
 {
 	printf("%s " PINGU_VERSION "\n", program);
