--- ../tests/UnifiedSpecTests/Operation.php	2026-09-07 18:49:41.124343370 +0300
+++ ../tests/UnifiedSpecTests/Operation.php	2026-09-07 18:52:44.708446016 +0300
@@ -20,6 +20,7 @@
 use MongoDB\Operation\FindOneAndUpdate;
 use PHPUnit\Framework\Assert;
 use PHPUnit\Framework\Constraint\IsType;
+use PHPUnit\Framework\NativeType;
 use PHPUnit\Framework\Exception as PHPUnitException;
 use stdClass;
 use Throwable;
@@ -483,7 +484,7 @@
                 assertArrayHasKey('filter', $args);
                 assertArrayHasKey('update', $args);
                 assertInstanceOf(stdClass::class, $args['filter']);
-                assertThat($args['update'], logicalOr(new IsType('array'), new IsType('object')));
+                assertThat($args['update'], logicalOr(new IsType(NativeType::Array), new IsType(NativeType::Object)));
 
                 return $collection->{$this->name}(
                     $args['filter'],
@@ -1056,7 +1057,7 @@
                     assertArrayHasKey('filter', $args);
                     assertArrayHasKey('update', $args);
                     assertInstanceOf(stdClass::class, $args['filter']);
-                    assertThat($args['update'], logicalOr(new IsType('array'), new IsType('object')));
+                    assertThat($args['update'], logicalOr(new IsType(NativeType::Array), new IsType(NativeType::Object)));
 
                     $bulk->{$type}(
                         $args['namespace'],
@@ -1122,7 +1123,7 @@
                 assertArrayHasKey('filter', $args);
                 assertArrayHasKey('update', $args);
                 assertInstanceOf(stdClass::class, $args['filter']);
-                assertThat($args['update'], logicalOr(new IsType('array'), new IsType('object')));
+                assertThat($args['update'], logicalOr(new IsType(NativeType::Array), new IsType(NativeType::Object)));
 
                 return [
                     $type => [
