diff --git a/src/configure.ml b/src/configure.ml
index f13d63e..f735a8c 100644
--- a/src/configure.ml
+++ b/src/configure.ml
@@ -7,7 +7,7 @@ let () =
   match Sys.argv with
   | [|_;"-cppo-args"|] ->
     let version = Scanf.sscanf Sys.ocaml_version "%d.%d." (fun major minor -> major * 100 + minor) in
-    printf "-D \\\"OCAML %d\\\"\n" version;
+    printf "-n -D \\\"OCAML %d\\\"\n" version;
     print_endline (if Sys.word_size = 32 then "-D WORD_SIZE_32 " else "");
     show_bytes "-D WITH_BYTES";
     exit 0
