--- ../tests/Common/Proxy/AutoloaderTest.php	2026-07-01 01:14:11.924216529 +0300
+++ ../tests/Common/Proxy/AutoloaderTest.php	2026-07-01 01:15:09.945831074 +0300
@@ -5,6 +5,7 @@
 use Doctrine\Common\Proxy\Autoloader;
 use Doctrine\Common\Proxy\Exception\InvalidArgumentException;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 use stdClass;
 use const DIRECTORY_SEPARATOR;
 use function class_exists;
@@ -28,9 +29,7 @@
         ];
     }
 
-    /**
-     * @dataProvider dataResolveFile
-     */
+    #[DataProvider('dataResolveFile')]
     public function testResolveFile($proxyDir, $proxyNamespace, $className, $expectedProxyFile)
     {
         $actualProxyFile = Autoloader::resolveFile($proxyDir, $proxyNamespace, $className);
