--- /dev/null		2018-12-14 23:39:56.564852402 +0200
+++ ../manage.py	2018-12-15 00:06:16.960041058 +0200
@@ -0,0 +1,11 @@
+#!/usr/bin/env python
+import os
+import sys
+
+if __name__ == "__main__":
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE",
+                          "crispy_forms.tests.test_settings")
+
+    from django.core.management import execute_from_command_line
+
+    execute_from_command_line(sys.argv)
