--- ../tests/DeepCopyTest/Matcher/PropertyTypeMatcherTest.php	2026-08-09 10:23:50.282596512 +0300
+++ ../tests/DeepCopyTest/Matcher/PropertyTypeMatcherTest.php	2026-08-09 10:41:15.230753243 +0300
@@ -5,6 +5,7 @@
 use DeepCopy\f009;
 use DeepCopy\Matcher\PropertyTypeMatcher;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 use stdClass;
 
 /**
@@ -12,9 +13,7 @@
  */
 class PropertyTypeMatcherTest extends TestCase
 {
-    /**
-     * @dataProvider providePairs
-     */
+    #[DataProvider('providePairs')]
     public function test_it_matches_the_given_property($object, $expected)
     {
         $matcher = new PropertyTypeMatcher(PropertyTypeMatcherTestFixture2::class);
@@ -49,7 +48,7 @@
         $this->assertTrue($matcher->matches($object, 'date'));
     }
 
-    public function providePairs()
+    public static function providePairs()
     {
         $object1 = new PropertyTypeMatcherTestFixture1();
         $object1->foo = new PropertyTypeMatcherTestFixture2();
