--- ../tests/ResponseTest.php	2026-07-09 23:15:51.454215665 +0300
+++ ../tests/ResponseTest.php	2026-07-09 23:17:15.232365158 +0300
@@ -4,6 +4,7 @@
 
 use Nyholm\Psr7\Response;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 use Psr\Http\Message\StreamInterface;
 
 /**
@@ -265,9 +266,7 @@
         ];
     }
 
-    /**
-     * @dataProvider trimmedHeaderValues
-     */
+    #[DataProvider('trimmedHeaderValues')]
     public function testHeaderValuesAreTrimmed($r)
     {
         $this->assertSame(['OWS' => ['Foo']], $r->getHeaders());
@@ -275,9 +274,7 @@
         $this->assertSame(['Foo'], $r->getHeader('OWS'));
     }
 
-    /**
-     * @dataProvider invalidWithHeaderProvider
-     */
+    #[DataProvider('invalidWithHeaderProvider')]
     public function testWithInvalidHeader($header, $headerValue, $expectedMessage): void
     {
         $r = new Response();
@@ -286,7 +283,7 @@
         $r->withHeader($header, $headerValue);
     }
 
-    public function invalidWithHeaderProvider(): iterable
+    public static function invalidWithHeaderProvider(): iterable
     {
         return [
             ['foo', [], 'Header values must be a string or an array of strings, empty array given'],
