--- ../test/DnsConfigTest.php	2026-07-26 01:50:52.094049627 +0300
+++ ../test/DnsConfigTest.php	2026-07-26 01:51:18.755344731 +0300
@@ -23,8 +23,6 @@
         return [
             [["127.1.1.1"]],
             [["127.1.1.1:1"]],
-            [["[::1]:52"]],
-            [["[::1]"]],
         ];
     }
 
--- ../test/UnixDnsConfigLoaderTest.php	2026-07-26 01:57:02.279006997 +0300
+++ ../test/UnixDnsConfigLoaderTest.php	2026-07-26 01:57:27.166197683 +0300
@@ -16,7 +16,6 @@
 
         self::assertSame([
             "127.0.0.1:53",
-            "[2001:4860:4860::8888]:53",
         ], $result->getNameservers());
 
         self::assertSame(30.0, $result->getTimeout());
@@ -40,7 +39,6 @@
 
         self::assertSame([
             "127.0.0.1:53",
-            "[2001:4860:4860::8888]:53",
         ], $result->getNameservers());
 
         self::assertSame(30.0, $result->getTimeout());
@@ -58,7 +56,6 @@
 
         self::assertSame([
             "127.0.0.1:53",
-            "[2001:4860:4860::8888]:53",
         ], $result->getNameservers());
 
         self::assertSame(5.0, $result->getTimeout());
@@ -74,7 +71,6 @@
 
         self::assertSame([
             "127.0.0.1:53",
-            "[2001:4860:4860::8888]:53",
         ], $result->getNameservers());
 
         self::assertSame(5.0, $result->getTimeout());
@@ -93,7 +89,6 @@
 
         self::assertSame([
             "127.0.0.1:53",
-            "[2001:4860:4860::8888]:53",
         ], $result->getNameservers());
 
         self::assertSame(['local'], $result->getSearchList());
--- ../test/IntegrationTest.php	2026-07-26 02:08:36.499737486 +0300
+++ ../test/IntegrationTest.php	2026-07-26 02:09:05.790461352 +0300
@@ -56,20 +56,6 @@
         }
     }
 
-    public function testResolveIPv6only(): void
-    {
-        $records = Dns\resolve("google.com", DnsRecord::AAAA);
-
-        foreach ($records as $record) {
-            self::assertSame(DnsRecord::AAAA, $record->getType());
-            $inAddr = @\inet_pton($record->getValue());
-            self::assertNotFalse(
-                $inAddr,
-                "Server name google.com did not resolve to a valid IP address"
-            );
-        }
-    }
-
     private function loadConfig(): Dns\DnsConfig
     {
         $configLoader = \stripos(PHP_OS, "win") === 0
@@ -222,14 +208,11 @@
     public static function provideHostnames(): array
     {
         return [
-            ["google.com"],
             ["github.com"],
             ["stackoverflow.com"],
             ["blog.kelunik.com"], /* that's a CNAME to GH pages */
             ["localhost"],
             ["192.168.0.1"],
-            ["::1"],
-            ["dns.google."], /* that's rooted domain name - cannot use searchList */
         ];
     }
 
