--- projects/rscriptd/Makefile.org	2013-01-18 18:30:04.000000000 +0200
+++ projects/rscriptd/Makefile	2013-01-18 18:31:40.000000000 +0200
@@ -67,7 +67,7 @@
 install-data:
 	# Install etc
 	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/rscriptd
-	install -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/rscriptd/rscriptd.conf
+	install -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/stargazer/rscriptd.conf
 ifeq ($(OS), linux)
 	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnConnect $(PREFIX)/etc/rscriptd/
 	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnDisconnect $(PREFIX)/etc/rscriptd/
