--- projects/rlm_stg/Makefile.org	2013-01-05 22:57:01.000000000 +0200
+++ projects/rlm_stg/Makefile	2013-01-05 22:57:14.000000000 +0200
@@ -57,9 +57,9 @@
 
 install-bin:
 ifeq ($(findstring $(DEFS), -DDEBUG),)
-	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/$(PROG)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/freeradius/$(PROG)
 else
-	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/$(PROG)
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/freeradius/$(PROG)
 endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
