--- projects/rscriptd/Makefile.org	2013-01-08 18:22:33.000000000 +0200
+++ projects/rscriptd/Makefile	2013-01-08 18:23:21.000000000 +0200
@@ -65,9 +65,9 @@
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 install-data:
-	install -D -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/rscriptd/rscriptd.conf
-	install -D -m $(BIN_MODE) -o $(OWNER) $(ETC_DIR_R)/OnConnect $(PREFIX)/etc/rscriptd/
-	install -D -m $(BIN_MODE) -o $(OWNER) $(ETC_DIR_R)/OnDisconnect $(PREFIX)/etc/rscriptd/
+	install -D -m $(DATA_MODE) -o $(OWNER) ./rscriptd.conf $(PREFIX)/etc/stargazer/rscriptd.conf
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnConnect $(PREFIX)/etc/stargazer/
+	install -D -m $(BIN_MODE) -o $(OWNER) ../stargazer/inst/linux/etc/stargazer/OnDisconnect $(PREFIX)/etc/stargazer/
 
 uninstall: uninstall-bin uninstall-data
 
