--- projects/rlm_stg/build.org	2013-01-08 20:09:37.000000000 +0200
+++ projects/rlm_stg/build	2013-01-08 20:09:51.000000000 +0200
@@ -16,18 +16,11 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-    else
-        MAKEOPTS="-j1"
-    fi
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgauth/build.org	2013-01-08 20:12:02.000000000 +0200
+++ projects/sgauth/build	2013-01-08 20:12:14.000000000 +0200
@@ -16,18 +16,11 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-    else
-        MAKEOPTS="-j1"
-    fi
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgconf/build.org	2013-01-08 20:12:43.000000000 +0200
+++ projects/sgconf/build	2013-01-08 20:12:54.000000000 +0200
@@ -16,18 +16,11 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-    else
-        MAKEOPTS="-j1"
-    fi
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgconf_xml/build.org	2013-01-08 20:13:24.000000000 +0200
+++ projects/sgconf_xml/build	2013-01-08 20:13:34.000000000 +0200
@@ -16,18 +16,11 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-    else
-        MAKEOPTS="-j1"
-    fi
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgconvertor/build.org	2013-01-08 20:14:04.000000000 +0200
+++ projects/sgconvertor/build	2013-01-08 20:14:29.000000000 +0200
@@ -16,8 +16,14 @@
 DIR_MODE=0755
 OWNER=root
 VAR_DIR="./inst/var/stargazer"
-DEFS="-DDEBUG"
-MAKEOPTS="-j1"
+
+if [ "$1" = "debug" ]
+then
+    DEFS="-DDEBUG"
+    MAKEOPTS="-j1"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+fi
+
 CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall -I/usr/local/include"
 LDFLAGS="$LDFLAGS -L/usr/local/lib"
 
--- projects/stargazer/build.org	2013-01-08 20:14:56.000000000 +0200
+++ projects/stargazer/build	2013-01-08 20:15:22.000000000 +0200
@@ -35,20 +35,11 @@
 XMLRPC_FEATURES="c++2 abyss-server"
 
 
-if [ -z "$1" ]
+if [ "$1" = "debug" ]
 then
-    DEFS="$DEFS -DNDEBUG"
+    DEFS="$DEFS -DDEBUG"
     MAKEOPTS="-j1"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="$DEFS -DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-    else
-        DEFS="$DEFS -DNDEBUG"
-        MAKEOPTS="-j1"
-    fi
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
@@ -410,9 +401,4 @@
 
 mkdir -p modules
 
-if [ "$1" != "debug" ]
-then
-    $MAKE $MAKEOPTS
-else
-    echo -e "\n\n\nDebug build. Type $MAKE explicitly"
-fi
+$MAKE $MAKEOPTS
--- projects/rscriptd/build.org	2013-01-09 11:47:23.000000000 +0200
+++ projects/rscriptd/build	2013-01-09 11:51:27.000000000 +0200
@@ -16,18 +16,11 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-    else
-        MAKEOPTS="-j1"
-    fi
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
@@ -37,6 +30,7 @@
 then
     OS=linux
     release=""
+    ETC_DIR_R="../stargazer/inst/linux/etc/stargazer"
     MAKE="make"
 fi
 
@@ -51,6 +45,7 @@
         9) OS=bsd7;;
         *) OS=unknown;;
     esac
+    ETC_DIR_R="../stargazer/inst/freebsd/etc/stargazer"
     MAKE="gmake"
 fi
 
@@ -171,6 +166,7 @@
 echo "DATA_MODE=$DATA_MODE" >> $CONFFILE
 echo "DIR_MODE=$DIR_MODE" >> $CONFFILE
 echo "OWNER=$OWNER" >> $CONFFILE
+echo "ETC_DIR_R=$ETC_DIR_R" >> $CONFFILE
 
 $MAKE $MAKEOPTS
 
