--- projects/stargazer/plugins/store/firebird/Makefile.org	2013-01-04 22:50:13.000000000 +0200
+++ projects/stargazer/plugins/store/firebird/Makefile	2013-01-04 22:50:56.000000000 +0200
@@ -22,5 +22,11 @@
 	  locker \
 	  crypto
 
+FB_CFLAGS = $(shell fb_config --cflags)
+FB_LDFLAGS = $(shell fb_config --libs)
+
+CXXFLAGS +=  $(FB_CFLAGS)
+LDFLAGS += $(FB_LDFLAGS)
+
 include ../../Makefile.in
 
--- projects/stargazer/plugins/Makefile.in.org	2013-01-04 23:07:57.000000000 +0200
+++ projects/stargazer/plugins/Makefile.in	2013-01-04 23:08:59.000000000 +0200
@@ -15,12 +15,12 @@
 
 CXXFLAGS += -fPIC $(STGLIBS_INCS)
 CFLAGS += -fPIC $(STGLIBS_INCS)
-LDFLAGS += -shared -Wl,-rpath,$(PREFIX)/usr/lib/stg $(STGLIBS_LIBS)
+LDFLAGS += -shared -Wl,-rpath,$(PREFIX)/usr/lib/stg
 
 all: $(PROG)
 
 $(PROG): $(OBJS)
-	$(CXX) $^ $(LDFLAGS) $(LIBS) -o $(PROG)
+	$(CXX) $^ $(STGLIBS_LIBS) $(LIBS) $(LDFLAGS) -o $(PROG)
 	$(LN) -fs "`pwd`/$(PROG)" $(DIR_MOD)/$(PROG)
 
 clean:
