--- projects/rscriptd/Makefile.org	2012-02-09 12:26:59.000000000 +0200
+++ projects/rscriptd/Makefile	2013-01-05 19:33:18.000000000 +0200
@@ -57,7 +57,11 @@
 install: install-bin install-data
 
 install-bin:
-	install -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 install-data:
--- projects/rlm_stg/Makefile.org	2012-02-09 12:26:58.000000000 +0200
+++ projects/rlm_stg/Makefile	2013-01-05 19:39:10.000000000 +0200
@@ -56,7 +56,11 @@
 install: install-bin
 
 install-bin:
-	install -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/$(PROG)
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/lib/freeradius/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/lib/freeradius/$(PROG)
+endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 uninstall: uninstall-bin
--- projects/sgauth/Makefile.org	2013-01-05 19:48:03.000000000 +0200
+++ projects/sgauth/Makefile	2013-01-05 19:49:54.000000000 +0200
@@ -58,7 +58,11 @@
 install: install-bin install-data
 
 install-bin:
-	install -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 install-data:
--- projects/sgconf/Makefile.org	2013-01-05 19:50:33.000000000 +0200
+++ projects/sgconf/Makefile	2013-01-05 19:51:15.000000000 +0200
@@ -66,7 +66,11 @@
 install: install-bin
 
 install-bin:
-	install -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/bin/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/bin/$(PROG)
+endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 uninstall: uninstall-bin
--- projects/sgconf_xml/Makefile.org	2013-01-05 19:52:02.000000000 +0200
+++ projects/sgconf_xml/Makefile	2013-01-05 19:52:38.000000000 +0200
@@ -66,7 +66,11 @@
 install: install-bin install-data
 
 install-bin:
-	install -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/bin/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/bin/$(PROG)
+endif
 	$(MAKE) -C $(DIR_LIBSRC) install
 
 uninstall: uninstall-bin uninstall-data
--- projects/stargazer/Makefile.org	2013-01-05 19:53:23.000000000 +0200
+++ projects/stargazer/Makefile	2013-01-05 19:54:28.000000000 +0200
@@ -77,7 +77,11 @@
 install: install-bin install-data
 
 install-bin: $(PROG)
-	install -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/sbin/$(PROG)
+endif
 	$(MAKE) -C $(DIR_INCLUDE) install
 	$(MAKE) -C $(DIR_LIBSRC) install
 	$(MAKE) -C $(DIR_PLUGINS) install
--- projects/convertor/Makefile.org	2013-01-05 20:00:32.000000000 +0200
+++ projects/convertor/Makefile	2013-01-05 20:02:51.000000000 +0200
@@ -58,6 +58,21 @@
 distclean: clean
 	rm -f ../../Makefile.conf
 
+install: install-bin
+
+install-bin:
+ifeq ($(findstring $(DEFS), -DDEBUG),)
+	install -D -m $(BIN_MODE) -o $(OWNER) $(PROG) $(PREFIX)/usr/bin/$(PROG)
+else
+	install -D -m $(BIN_MODE) -o $(OWNER) -s $(PROG) $(PREFIX)/usr/bin/$(PROG)
+endif
+	$(MAKE) -C $(DIR_PLUGINS) install
+
+uninstall: uninstall-bin
+
+uninstall-bin:
+	rm -f $(PREFIX)/usr/bin/$(PROG)
+
 ifneq ($(MAKECMDGOALS),distclean)
 ifneq ($(MAKECMDGOALS),clean)
 ifneq ($(MAKECMDGOALS),uninstall)
