--- projects/sgauth/Makefile.org	2013-01-05 23:47:38.000000000 +0200
+++ projects/sgauth/Makefile	2013-01-05 23:48:34.000000000 +0200
@@ -67,8 +67,8 @@
 
 install-data:
 	# Install etc
-	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc
-	install -m $(DATA_MODE) -o $(OWNER) ./sgauth.conf $(PREFIX)/etc/sgauth.conf
+	mkdir -m $(DIR_MODE) -p $(PREFIX)/etc/stargazer
+	install -m $(DATA_MODE) -o $(OWNER) ./sgauth.conf $(PREFIX)/etc/stargazer/sgauth.conf
 
 uninstall: uninstall-bin uninstall-data
 
