--- projects/rlm_stg/build.org	2013-01-12 16:16:43.000000000 +0200
+++ projects/rlm_stg/build	2013-01-12 16:19:03.000000000 +0200
@@ -16,21 +16,14 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-    DEBUG="no"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
 else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-        DEBUG="yes"
-    else
-        MAKEOPTS="-j1"
-        DEBUG="no"
-    fi
+    DEBUG="no"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/rscriptd/build.org	2013-01-12 16:19:53.000000000 +0200
+++ projects/rscriptd/build	2013-01-12 16:20:13.000000000 +0200
@@ -16,21 +16,14 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-    DEBUG="no"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
 else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-        DEBUG="yes"
-    else
-        MAKEOPTS="-j1"
-        DEBUG="no"
-    fi
+    DEBUG="no"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgauth/build.org	2013-01-12 16:20:34.000000000 +0200
+++ projects/sgauth/build	2013-01-12 16:20:54.000000000 +0200
@@ -16,21 +16,14 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-    DEBUG="no"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
 else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-        DEBUG="yes"
-    else
-        MAKEOPTS="-j1"
-        DEBUG="no"
-    fi
+    DEBUG="no"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgconf/build.org	2013-01-12 16:21:04.000000000 +0200
+++ projects/sgconf/build	2013-01-12 16:21:18.000000000 +0200
@@ -16,21 +16,14 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-    DEBUG="no"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
 else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-        DEBUG="yes"
-    else
-        MAKEOPTS="-j1"
-        DEBUG="no"
-    fi
+    DEBUG="no"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgconf_xml/build.org	2013-01-12 16:21:26.000000000 +0200
+++ projects/sgconf_xml/build	2013-01-12 16:21:42.000000000 +0200
@@ -16,21 +16,14 @@
 DIR_MODE=0755
 OWNER=root
 
-if [ -z $1 ]
+if [ "$1" = "debug" ]
 then
+    DEFS="-DDEBUG"
     MAKEOPTS="-j1"
-    DEBUG="no"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
 else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="-DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-        DEBUG="yes"
-    else
-        MAKEOPTS="-j1"
-        DEBUG="no"
-    fi
+    DEBUG="no"
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
--- projects/sgconv/build.org	2013-01-12 16:21:51.000000000 +0200
+++ projects/sgconv/build	2013-01-12 16:24:26.000000000 +0200
@@ -16,11 +16,19 @@
 DIR_MODE=0755
 OWNER=root
 VAR_DIR="./inst/var/stargazer"
-DEFS="-DDEBUG"
-MAKEOPTS="-j1"
+
+if [ "$1" = "debug" ]
+then
+    DEFS="-DDEBUG"
+    MAKEOPTS="-j1"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
+else
+    DEBUG="no"
+fi
+
 CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall -I/usr/local/include"
 LDFLAGS="$LDFLAGS -L/usr/local/lib"
-DEBUG="yes"
 
 if [ "$sys" = "Linux" ]
 then
--- projects/stargazer/build.org	2013-01-12 16:27:47.000000000 +0200
+++ projects/stargazer/build	2013-01-12 16:28:52.000000000 +0200
@@ -34,24 +34,16 @@
 MIN_XMLRPCC_VERSION="1.06.27"
 XMLRPC_FEATURES="c++2 abyss-server"
 
-
-if [ -z "$1" ]
+if [ "$1" = "debug" ]
 then
+    DEFS="$DEFS -DDEBUG"
+    MAKEOPTS="-j1"
+    CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
+    DEBUG="yes"
+else
     DEFS="$DEFS -DNDEBUG"
     MAKEOPTS="-j1"
     DEBUG="no"
-else
-    if [ "$1" = "debug" ]
-    then
-        DEFS="$DEFS -DDEBUG"
-        MAKEOPTS="-j1"
-        CXXFLAGS="$CXXFLAGS -ggdb3 -W -Wall"
-        DEBUG="yes"
-    else
-        DEFS="$DEFS -DNDEBUG"
-        MAKEOPTS="-j1"
-        DEBUG="no"
-    fi
 fi
 
 CXXFLAGS="$CXXFLAGS -I/usr/local/include"
