--- ../tests/HostsFile/HostsFileTest.php	2026-08-12 11:54:05.866433345 +0300
+++ ../tests/HostsFile/HostsFileTest.php	2026-08-12 11:55:34.588146803 +0300
@@ -23,10 +23,6 @@
                 self::BOTH_RECORD['name'] => IPv4Address::parse(self::BOTH_RECORD['v4']),
                 self::IPV4_ONLY_RECORD['name'] => IPv4Address::parse(self::IPV4_ONLY_RECORD['v4']),
             ],
-            ResourceTypes::AAAA => [
-                self::BOTH_RECORD['name'] => IPv4Address::parse(self::BOTH_RECORD['v6']),
-                self::IPV6_ONLY_RECORD['name'] => IPv4Address::parse(self::IPV6_ONLY_RECORD['v6']),
-            ],
         ];
     }
 
@@ -36,12 +32,10 @@
 
         $this->assertFalse($hostsFile->containsName(self::NON_EXISTENT_NAME));
         $this->assertFalse($hostsFile->containsName(self::NON_EXISTENT_NAME, ResourceTypes::A));
-        $this->assertFalse($hostsFile->containsName(self::NON_EXISTENT_NAME, ResourceTypes::AAAA));
 
         $name = DomainName::fromString(self::NON_EXISTENT_NAME);
         $this->assertFalse($hostsFile->containsName($name));
         $this->assertFalse($hostsFile->containsName($name, ResourceTypes::A));
-        $this->assertFalse($hostsFile->containsName($name, ResourceTypes::AAAA));
     }
 
     public function testContainsNameWithInvalidName()
@@ -50,7 +44,6 @@
 
         $this->assertFalse($hostsFile->containsName(self::INVALID_NAME));
         $this->assertFalse($hostsFile->containsName(self::INVALID_NAME, ResourceTypes::A));
-        $this->assertFalse($hostsFile->containsName(self::INVALID_NAME, ResourceTypes::AAAA));
     }
 
     public function testContainsNameWithIPv4Only()
@@ -59,26 +52,10 @@
 
         $this->assertTrue($hostsFile->containsName(self::IPV4_ONLY_RECORD['name']));
         $this->assertTrue($hostsFile->containsName(self::IPV4_ONLY_RECORD['name'], ResourceTypes::A));
-        $this->assertFalse($hostsFile->containsName(self::IPV4_ONLY_RECORD['name'], ResourceTypes::AAAA));
 
         $name = DomainName::fromString(self::IPV4_ONLY_RECORD['name']);
         $this->assertTrue($hostsFile->containsName($name));
         $this->assertTrue($hostsFile->containsName($name, ResourceTypes::A));
-        $this->assertFalse($hostsFile->containsName($name, ResourceTypes::AAAA));
-    }
-
-    public function testContainsNameWithIPv6Only()
-    {
-        $hostsFile = new HostsFile($this->provideData());
-
-        $this->assertTrue($hostsFile->containsName(self::IPV6_ONLY_RECORD['name']));
-        $this->assertFalse($hostsFile->containsName(self::IPV6_ONLY_RECORD['name'], ResourceTypes::A));
-        $this->assertTrue($hostsFile->containsName(self::IPV6_ONLY_RECORD['name'], ResourceTypes::AAAA));
-
-        $name = DomainName::fromString(self::IPV6_ONLY_RECORD['name']);
-        $this->assertTrue($hostsFile->containsName($name));
-        $this->assertFalse($hostsFile->containsName($name, ResourceTypes::A));
-        $this->assertTrue($hostsFile->containsName($name, ResourceTypes::AAAA));
     }
 
     public function testContainsNameWithBoth()
@@ -87,12 +64,10 @@
 
         $this->assertTrue($hostsFile->containsName(self::BOTH_RECORD['name']));
         $this->assertTrue($hostsFile->containsName(self::BOTH_RECORD['name'], ResourceTypes::A));
-        $this->assertTrue($hostsFile->containsName(self::BOTH_RECORD['name'], ResourceTypes::AAAA));
 
         $name = DomainName::fromString(self::BOTH_RECORD['name']);
         $this->assertTrue($hostsFile->containsName($name));
         $this->assertTrue($hostsFile->containsName($name, ResourceTypes::A));
-        $this->assertTrue($hostsFile->containsName($name, ResourceTypes::AAAA));
     }
 
     public function testGetAddressForNameWithNonExistentNameDefaultFamily()
@@ -105,11 +80,6 @@
         $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::NON_EXISTENT_NAME, ResourceTypes::A));
     }
 
-    public function testGetAddressForNameWithNonExistentNameIPv6()
-    {
-        $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::NON_EXISTENT_NAME, ResourceTypes::AAAA));
-    }
-
     public function testGetAddressForNameWithInvalidNameDefaultFamily()
     {
         $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::INVALID_NAME));
@@ -120,10 +90,6 @@
         $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::INVALID_NAME, ResourceTypes::A));
     }
 
-    public function testGetAddressForNameWithInvalidNameIPv6()
-    {
-        $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::INVALID_NAME, ResourceTypes::AAAA));
-    }
 
     public function testGetAddressForNameWithExistingIPv4()
     {
@@ -138,18 +104,6 @@
         $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::IPV6_ONLY_RECORD['name'], ResourceTypes::A));
     }
 
-    public function testGetAddressForNameWithExistingIPv6()
-    {
-        $hostsFile = new HostsFile($this->provideData());
-
-        $this->assertSame(self::IPV6_ONLY_RECORD['v6'], (string)$hostsFile->getAddressForName(self::IPV6_ONLY_RECORD['name'], ResourceTypes::AAAA));
-    }
-
-    public function testGetAddressForNameWithNonExistentIPv6()
-    {
-        $this->assertNull((new HostsFile($this->provideData()))->getAddressForName(self::IPV4_ONLY_RECORD['name'], ResourceTypes::AAAA));
-    }
-
     public function testToArray()
     {
         $data = $this->provideData();
