--- a/Makefile	2017-07-30 21:44:54.000000000 +0300
+++ b/Makefile	2017-07-30 22:06:50.000000000 +0300
@@ -72,9 +72,7 @@
 install-local: doc-install
 ifeq ($(INSTALL_DAEMON),yes)
 	$(INSTALL_DIRECTORY) -o ${user} -g ${group} -m 2775 \
-	  ${localstatedir}/log/liquidsoap
-	$(INSTALL_DIRECTORY) -o ${user} -g ${group} -m 2775 \
-	  ${localstatedir}/run/liquidsoap
+	  ${DESTDIR}/var/log/liquidsoap
 endif
 	$(INSTALL_DIRECTORY) $(bindir)
 	$(INSTALL_DIRECTORY) $(libdir)/liquidsoap
