--- ../Makefile	2019-08-18 19:05:20.612623767 +0300
+++ ../Makefile	2019-08-18 19:05:52.215695031 +0300
@@ -68,8 +68,7 @@
 
 install-local: doc-install
 ifeq ($(INSTALL_DAEMON),yes)
-	$(INSTALL_DIRECTORY) -o ${user} -g ${group} -m 2775 ${localstatedir}/log/liquidsoap
-	$(INSTALL_DIRECTORY) -o ${user} -g ${group} -m 2775 ${localstatedir}/run/liquidsoap
+	$(INSTALL_DIRECTORY) -o ${user} -g ${group} -m 2775 ${DESTDIR}/var/log/liquidsoap
 endif
 	$(INSTALL_DIRECTORY) $(bindir)
 	$(INSTALL_DIRECTORY) $(libdir)/liquidsoap/$(libs_dir_version)
