--- a/conf/internet.conf	2016-05-10 17:43:59.021848960 +0300
+++ b/conf/internet.conf	2016-05-10 17:44:13.371849599 +0300
@@ -42,6 +42,6 @@
 #
 #  ИЛИ
 #
-#include :/usr/local/accnt/bin/ipcadcnt c f 2 2
+#include :/opt/topola/bin/ipcadcnt c f 2 2
 
 #####################################################
--- a/conf/iptstat.conf	2016-05-10 17:44:33.256850484 +0300
+++ b/conf/iptstat.conf	2016-05-10 17:44:54.812851443 +0300
@@ -45,7 +45,7 @@
 # Флаг-файл удобно использовать в скриптах, как подсказку
 # для создания цепочек счетчиков.
 
-iptflag /usr/local/accnt/conf/iptcount.flg
+iptflag /opt/topola/conf/iptcount.flg
 
 ########################################################
 # Эта опция влючает дополнительную функцию коллектора.
@@ -61,6 +61,6 @@
 # случае программа просто создаст общий список.
 #
 
-nodefile /usr/local/accnt/conf/internet.lst
+nodefile /opt/topola/conf/internet.lst
 
 ########################################################
