--- ../tpafunc	2021-03-06 16:48:33.724799549 +0200
+++ ../tpafunc	2021-03-06 16:49:30.898918076 +0200
@@ -131,7 +131,6 @@
 
 fi
 
-HOME_DIR=`grep -a ^${TPA_OWNER}: /etc/passwd | cut -d: -f6`
 
 if [ -z "${TPA_HOME}" ]; then
   while [ -z "${T_HOME}" ]; do
@@ -152,9 +151,6 @@
   export TPA_HOME=${T_HOME}
 fi
 
-  if [ -f ${TPA_HOME}/bin/topola.conf ]; then
-    D_DIR=`grep -a ^"data_dir " ${TPA_HOME}/bin/topola.conf | cut -d" " -f2`
-  fi
   [ -z "$D_DIR" ] && D_DIR=${TPA_HOME}/data
 
 export DATA_DIR=${D_DIR}
@@ -185,9 +181,6 @@
   fi
   . ${HOME}/.topola
 
-  if [ -f ${TPA_HOME}/bin/topola.conf ]; then
-    D_DIR=`grep -a ^"data_dir " ${TPA_HOME}/bin/topola.conf | cut -d" " -f2`
-  fi
   [ -z "$D_DIR" ] && D_DIR=${TPA_HOME}/data
 
   if [ ! -f $D_DIR/main/system.conf ]; then
@@ -231,7 +224,7 @@
 
   [ ! -f "${TPA_HOME}/$1" ] && return 0
 
-  sed -e s\#@@TPAHOME@@#${TPA_HOME}#g -e s\#@@TPAOWNER@@#${TPA_OWNER}#g ${TPA_HOME}/$1 > ${TPA_HOME}/$1.aaa
+  sed -e s\#@@TPAHOME@@#${TPAHOME}#g -e s\#@@TPAOWNER@@#${TPA_OWNER}#g ${TPA_HOME}/$1 > ${TPA_HOME}/$1.aaa
   mv ${TPA_HOME}/$1.aaa ${TPA_HOME}/$1
 
   [ "$SGN_TPA" = "yes" ] && make_sign_tpa ${TPA_HOME}/$1
