--- a/tpainst.sh	2013-09-07 15:19:10.000000000 +0300
+++ b/tpainst.sh	2016-05-10 12:17:03.894217002 +0300
@@ -12,7 +12,7 @@
 make_dir src || exit 1
 
 # 2. Install binaries
-sed s.TAREM_DIR=/usr/local/accnt.TAREM_DIR=${TPA_HOME}. src/Makefile > src/aaa
+sed 's.TAREM_DIR=/usr/local/accnt.TAREM_DIR=${TPAHOME}.' src/Makefile > src/aaa
 mv src/aaa src/Makefile
 
 if which gmake >/dev/null 2>&1 ; then
