--- ../tpafunc	2021-03-06 16:12:54.026575813 +0200
+++ ../tpafunc	2021-03-06 16:13:06.385626463 +0200
@@ -136,7 +136,6 @@
 
 fi
 
-HOME_DIR=`grep -a ^${TPA_OWNER}: /etc/passwd | cut -d: -f6`
 
 if [ -z "${TPA_HOME}" ]; then
   while [ -z "${T_HOME}" ]; do
@@ -168,7 +167,7 @@
 
 # export TPA_DOMAIN
 
-echo export TPA_HOME=${TPA_HOME}            > ${HOME_DIR}/.topola
+echo export TPA_HOME=${TPAHOME}            > ${HOME_DIR}/.topola
 echo export TPA_OWNER=${TPA_OWNER}         >> ${HOME_DIR}/.topola
 echo export TPA_INSTALLER=${TPA_INSTALLER} >> ${HOME_DIR}/.topola
 echo export TPA_DOMAIN=${TPA_DOMAIN}       >> ${HOME_DIR}/.topola
 
@@ -224,7 +224,7 @@
 
   [ ! -f "${TPA_HOME}/$1" ] && return 0
 
-  sed -e s\#@@TPAHOME@@#${TPA_HOME}#g -e s\#@@TPAOWNER@@#${TPA_OWNER}#g ${TPA_HOME}/$1 > ${TPA_HOME}/$1.aaa
+  sed -e s\#@@TPAHOME@@#${TPAHOME}#g -e s\#@@TPAOWNER@@#${TPA_OWNER}#g ${TPA_HOME}/$1 > ${TPA_HOME}/$1.aaa
   mv ${TPA_HOME}/$1.aaa ${TPA_HOME}/$1
 
   [ "$SGN_TPA" = "yes" ] && make_sign_tpa ${TPA_HOME}/$1
@@ -285,7 +285,7 @@
     fi
   fi
   
-  [ ${TPA_OWNER} = ${TPA_INSTALLER} ] && return 0
+  [ "${TPA_OWNER}" = "${TPA_INSTALLER}" ] && return 0
   
   chown -R "${TPA_OWNER}" ${MMKD} > /dev/null 2>&1
   if [ $? -ne 0 ]; then 
@@ -312,7 +312,7 @@
      prn_warn_mark "Unable to move at $MMVD" ; return 1
   fi
 
-  if [ ${TPA_OWNER} != ${TPA_INSTALLER} ]; then
+  if [ "${TPA_OWNER}" != "${TPA_INSTALLER}" ]; then
     chown -R "${TPA_OWNER}" ${MMVD}
     if [ $? -ne 0 ]; then 
        prn_warn_mark "Unable to chmod $MMVD" ; return 1
@@ -367,7 +367,7 @@
   if [ $? -ne 0 ]; then
     prn_err_mark "Unable move $1 to binary dir" ; return 1
   fi
-  if [ ${TPA_OWNER} != ${TPA_INSTALLER} ]; then
+  if [ "${TPA_OWNER}" != "${TPA_INSTALLER}" ]; then
     chown -R "${TPA_OWNER}" ${TPA_HOME}/bin/$1
     if [ $? -ne 0 ]; then 
       prn_warn_mark "Unable to chown" ; return 1
@@ -403,7 +403,7 @@
   if [ $? -ne 0 ]; then
     prn_err_mark "Unable move $1 to dso dir" ; return 1
   fi
-  if [ ${TPA_OWNER} != ${TPA_INSTALLER} ]; then
+  if [ "${TPA_OWNER}" != "${TPA_INSTALLER}" ]; then
     chown -R "${TPA_OWNER}" ${TPA_HOME}/bin/dso/$1
     if [ $? -ne 0 ]; then 
       prn_warn_mark "Unable to chown" ; return 1
