--- ./mongomock/collection.py	2025-10-07 23:51:19.648517698 +0300
+++ ./mongomock/collection.py	2025-10-07 23:55:14.307470328 +0300
@@ -189,14 +189,14 @@
             return ret_val
         self.builder.executors.append(exec_update)
 
-    def update(self, document, hint=None):
-        self.register_update_op(document, multi=True, hint=hint)
+    def update(self, document, hint=None, sort=None):
+        self.register_update_op(document, multi=True, hint=hint, sort=sort)
 
-    def update_one(self, document, hint=None):
-        self.register_update_op(document, multi=False, hint=hint)
+    def update_one(self, document, hint=None, sort=None):
+        self.register_update_op(document, multi=False, hint=hint, sort=sort)
 
-    def replace_one(self, document, hint=None):
-        self.register_update_op(document, multi=False, remove=True, hint=hint)
+    def replace_one(self, document, hint=None, sort=None):
+        self.register_update_op(document, multi=False, remove=True, hint=hint, sort=sort)
 
 
 def _combine_projection_spec(projection_fields_spec):
@@ -392,16 +392,16 @@
         self.insert(doc)
 
     def add_update(self, selector, doc, multi=False, upsert=False, collation=None,
-                   array_filters=None, hint=None):
+                   array_filters=None, hint=None, sort=None):
         if array_filters:
             raise_not_implemented(
                 'array_filters', 'Array filters are not implemented in mongomock yet.')
         write_operation = BulkWriteOperation(self, selector, is_upsert=upsert)
-        write_operation.register_update_op(doc, multi, hint=hint)
+        write_operation.register_update_op(doc, multi, hint=hint, sort=sort)
 
-    def add_replace(self, selector, doc, upsert, collation=None, hint=None):
+    def add_replace(self, selector, doc, upsert, collation=None, hint=None, sort=None):
         write_operation = BulkWriteOperation(self, selector, is_upsert=upsert)
-        write_operation.replace_one(doc, hint=hint)
+        write_operation.replace_one(doc, hint=hint, sort=sort)
 
     def add_delete(self, selector, just_one, collation=None, hint=None):
         write_operation = BulkWriteOperation(self, selector, is_upsert=False)
@@ -609,13 +609,13 @@
 
     def update_one(
             self, filter, update, upsert=False, bypass_document_validation=False, collation=None,
-            array_filters=None, hint=None, session=None, let=None):
+            array_filters=None, hint=None, session=None, let=None, sort=None):
         if not bypass_document_validation:
             validate_ok_for_update(update)
         return UpdateResult(
             self._update(
                 filter, update, upsert=upsert, hint=hint, session=session, collation=collation,
-                array_filters=array_filters, let=let),
+                array_filters=array_filters, let=let, sort=sort),
             acknowledged=True)
 
     def update_many(
@@ -632,11 +632,11 @@
 
     def replace_one(
             self, filter, replacement, upsert=False, bypass_document_validation=False,
-            session=None, hint=None):
+            session=None, hint=None, sort=None):
         if not bypass_document_validation:
             validate_ok_for_replace(replacement)
         return UpdateResult(
-            self._update(filter, replacement, upsert=upsert, hint=hint, session=session),
+            self._update(filter, replacement, upsert=upsert, hint=hint, session=session, sort=sort),
             acknowledged=True)
 
     if helpers.PYMONGO_VERSION < version.parse('4.0'):
