--- ../tests/ClosureExpressionVisitorTest.php	2026-07-08 00:11:18.516821714 +0300
+++ ../tests/ClosureExpressionVisitorTest.php	2026-07-08 00:11:35.475443630 +0300
@@ -49,7 +49,7 @@
         self::assertSame(42, $this->visitor->getObjectFieldValue($object, 'foo.fooBar', true));
     }
 
-    #[RequiresPhp('>= 8.4')]
+    #[RequiresPhp('>= 8.4.0')]
     public function testGetObjectFieldValueAccessingRawValueBypassingPropertyHook(): void
     {
         $object         = new TestObjectPropertyHook();
