--- ../src/crypto/ms_srtp.cpp	2025-06-22 21:11:10.615903165 +0300
+++ ../src/crypto/ms_srtp.cpp	2025-06-22 21:12:35.024079931 +0300
@@ -1700,7 +1700,7 @@
 }
 
 extern "C" int ms_media_stream_sessions_set_srtp_recv_key(
-    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const char *key, size_t key_length, MSSrtpKeySource source) {
+    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const uint8_t *key, size_t key_length, MSSrtpKeySource source) {
 	ms_error("Unable to set srtp recv key: srtp support disabled in mediastreamer2");
 	return -1;
 }
@@ -1714,7 +1714,7 @@
 }
 
 extern "C" int ms_media_stream_sessions_set_srtp_send_key(
-    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const char *key, size_t key_length, MSSrtpKeySource source) {
+    MSMediaStreamSessions *sessions, MSCryptoSuite suite, const uint8_t *key, size_t key_length, MSSrtpKeySource source) {
 	ms_error("Unable to set srtp send key: srtp support disabled in mediastreamer2");
 	return -1;
 }
@@ -1757,7 +1757,7 @@
 	ms_error("Unable to set EKT operation mode: srtp support disabled in mediastreamer2");
 	return -1;
 }
-extern "C" int ms_media_stream_sessions_set_ekt(MSMediaStreamSessions *sessions, MSEKTParametersSet *ekt) {
+extern "C" int ms_media_stream_sessions_set_ekt(MSMediaStreamSessions *sessions, const MSEKTParametersSet *ekt_params) {
 	ms_error("Unable to set EKT key: srtp support disabled in mediastreamer2");
 	return -1;
 }
