--- ../xt_NAT.c	2021-02-13 21:20:13.102284452 +0200
+++ ../xt_NAT.c	2021-02-13 21:21:15.444180778 +0200
@@ -1014,8 +1014,8 @@
             if (unlikely(skb_shinfo(skb)->nr_frags > 1 && skb_headlen(skb) == sizeof(struct iphdr))) {
                 frag = &skb_shinfo(skb)->frags[0];
                 //printk(KERN_DEBUG "xt_NAT DNAT: frag_size = %d (required %lu)\n", frag->size, sizeof(struct tcphdr));
-                if (unlikely(frag->size < sizeof(struct tcphdr))) {
-                        printk(KERN_DEBUG "xt_NAT DNAT: drop TCP frag_size = %d\n", frag->size);
+                if (unlikely(skb_frag_size(frag) < sizeof(struct tcphdr))) {
+                        printk(KERN_DEBUG "xt_NAT DNAT: drop TCP frag_size = %d\n", skb_frag_size(frag));
                         return NF_DROP;
                 }
                 tcp = (struct tcphdr *)skb_frag_address_safe(frag);
@@ -1096,8 +1096,8 @@
             if (unlikely(skb_shinfo(skb)->nr_frags > 1 && skb_headlen(skb) == sizeof(struct iphdr))) {
                 frag = &skb_shinfo(skb)->frags[0];
                 //printk(KERN_DEBUG "xt_NAT DNAT: frag_size = %d (required %lu)\n", frag->size, sizeof(struct udphdr));
-                if (unlikely(frag->size < sizeof(struct udphdr))) {
-                        printk(KERN_DEBUG "xt_NAT DNAT: drop UDP frag_size = %d\n", frag->size);
+                if (unlikely(skb_frag_size(frag) < sizeof(struct udphdr))) {
+                        printk(KERN_DEBUG "xt_NAT DNAT: drop UDP frag_size = %d\n", skb_frag_size(frag));
                         return NF_DROP;
                 }
                 udp = (struct udphdr *)skb_frag_address_safe(frag);
