--- ../src/Context/ThreadContext.php	2026-08-29 23:44:57.711090161 +0300
+++ ../src/Context/ThreadContext.php	2026-08-29 23:45:27.914479833 +0300
@@ -70,6 +70,7 @@
             $paths = [
                 \dirname(__DIR__, 2) . \DIRECTORY_SEPARATOR . "vendor" . \DIRECTORY_SEPARATOR . "autoload.php",
                 \dirname(__DIR__, 4) . \DIRECTORY_SEPARATOR . "autoload.php",
+                \dirname(__DIR__) . \DIRECTORY_SEPARATOR . "autoload.php",
             ];
 
             foreach ($paths as $path) {
