--- ../phpunit.xml.dist	2026-07-26 00:22:16.550854661 +0300
+++ ../phpunit.xml.dist	2026-07-26 00:23:38.627635732 +0300
@@ -15,6 +15,9 @@
     <testsuites>
         <testsuite name="Amp Concurrent">
             <directory>test</directory>
+            <exclude>test/Context/AbstractContextTest.php</exclude>
+            <exclude>test/Worker/AbstractPoolTest.php</exclude>
+            <exclude>test/Worker/AbstractWorkerTest.php</exclude>
         </testsuite>
     </testsuites>
     <coverage>
