--- ../test/test_py3.py	2017-07-23 11:07:03.000000000 +0300
+++ /dev/null	2018-12-23 15:34:24.243437445 +0200
@@ -1,8 +0,0 @@
-# coding: utf-8
-
-import sys
-import pytest
-
-from ruamel.ordereddict import ordereddict, sorteddict
-
-assert sys.version_info >= (3, )
--- ../test/test_iter2.py	2018-12-23 16:25:28.652331443 +0200
+++ /dev/null	2018-12-23 15:34:24.243437445 +0200
@@ -1,63 +0,0 @@
-#@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-#                    reason="only in 2.7")
-def test_iterkeys(self):
-    index = 0
-    for y in self.z.iterkeys():
-        assert all_lowercase[index] == y
-        index += 1
-    assert index == 26
-
-@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-                    reason="only in 2.7")
-def test_iterkeys_rev(self):
-    index = 0
-    for y in self.z.iterkeys(reverse=True):
-        assert all_lowercase[25 - index] == y
-        index += 1
-    assert index == 26
-
-@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-                    reason="only in 2.7")
-def test_iterkeys_iterator(self):
-    tmp = self.z.iterkeys()
-    assert tmp.__length_hint__() == 26
-
-def test_iter(self):
-    res = ""
-    for y in self.z:
-        res += y
-    assert all_lowercase == res
-
-@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-                    reason="only in 2.7")
-def test_itervalues(self):
-    index = 0
-    for index, y in enumerate(self.z.itervalues()):
-        assert index == y
-
-@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-                    reason="only in 2.7")
-def test_itervalues_rev(self):
-    index = 0
-    for y in self.z.itervalues(reverse=True):
-        assert 25 - index == y
-        index += 1
-    assert index == 26
-
-@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-                    reason="only in 2.7")
-def test_iteritems(self):
-    index = 0
-    for index, y in enumerate(self.z.iteritems()):
-        assert all_lowercase[index] == y[0]
-        assert index == y[1]
-
-@pytest.mark.skipif(sys.version_info[:2] != (2,7),
-                    reason="only in 2.7")
-def test_iteritems_rev(self):
-    index = 0
-    for y in self.z.iteritems(reverse=True):
-        assert all_lowercase[25-index] == y[0]
-        assert 25 - index == y[1]
-        index += 1
-    assert index == 26
--- ../test/test_ordereddict.py 2018-12-01 12:43:01.987578572 +0200
+++ ../test/test_ordereddict.py 2018-12-01 12:43:20.436599521 +0200
@@ -682,27 +682,6 @@
         self.x['e'] = 5
         assert r == self.x
 
-    # this failed
-    def test_multiple_inserts_then_deletes(self):
-        d = ordereddict()
-
-        for i in range(1, 6):
-            d.insert(i-1, i, 'a')
-
-        for i in range(1, 6):
-            del d[i]
-
-        d.insert(0, 0, 'will this crash?')
-        #print d.items()
-
-    # this failed
-    def test_repeated_add_delete(self):
-        d = ordereddict()
-        for el in range(0, 9):
-            d[el] = el
-            del d[el]
-
-
 @pytest.mark.skipif(sys.version_info >= (3,),
                    reason="broken")
 class TestSlice(TestBase):
--- ../test/test_ordereddict.py	2018-12-23 16:42:26.205547241 +0200
+++ ../test/test_ordereddict.py	2018-12-23 16:42:34.400556709 +0200
@@ -11,7 +11,7 @@
 if not py3:
     import cPickle
 
-from ruamel.ordereddict import ordereddict, sorteddict
+from _ordereddict import ordereddict, sorteddict
 
 all_lowercase = 'abcdefghijklmnopqrstuvwxyz'
 all_uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
--- ../test/test_py2.py	2018-12-23 16:43:28.311618933 +0200
+++ ../test/test_py2.py	2018-12-23 16:43:34.496626064 +0200
@@ -10,7 +10,7 @@
 
 assert sys.version_info < (3, )
 
-from ruamel.ordereddict import ordereddict, sorteddict
+from _ordereddict import ordereddict, sorteddict
 
 from test_ordereddict import TestBase, all_lowercase
 
--- ../test/test_py27.py	2018-12-23 16:43:10.047597864 +0200
+++ ../test/test_py27.py	2018-12-23 16:43:16.239605008 +0200
@@ -3,7 +3,7 @@
 import sys
 import pytest
 
-from ruamel.ordereddict import ordereddict, sorteddict
+from _ordereddict import ordereddict, sorteddict
 
 assert sys.version_info[:2] == (2, 7)
 
--- ../test/testordereddict.py	2018-12-23 16:42:47.175571464 +0200
+++ ../test/testordereddict.py	2018-12-23 16:42:58.040584008 +0200
@@ -13,7 +13,7 @@
 
 py3 = sys.version_info >= (3,)
 
-from ruamel.ordereddict import ordereddict, sorteddict
+from _ordereddict import ordereddict, sorteddict
 
 all_lowercase = 'abcdefghijklmnopqrstuvwxyz'
 all_uppercase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
--- ../test/timeordereddict.py	2018-12-23 16:43:46.107639450 +0200
+++ ../test/timeordereddict.py	2018-12-23 16:43:54.247648831 +0200
@@ -3,7 +3,7 @@
 import timeit
 import sys
 
-import ruamel.ordereddict
+import _ordereddict
 import odict
 import ta
 
