--- ../test/JsonSerializerTest.php	2026-07-10 20:45:56.884148545 +0300
+++ ../test/JsonSerializerTest.php	2026-07-10 20:46:51.385546968 +0300
@@ -6,6 +6,7 @@
 use Amp\Serialization\SerializationException;
 use Amp\Serialization\Serializer;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 class JsonSerializerTest extends TestCase
 {
@@ -16,7 +17,7 @@
         return JsonSerializer::withAssociativeArrays($encodeOptions, $decodeOptions);
     }
 
-    public function provideSerializableData(): iterable
+    public static function provideSerializableData(): iterable
     {
         return [
             ['test'],
@@ -27,9 +28,7 @@
         ];
     }
 
-    /**
-     * @dataProvider provideSerializableData
-     */
+    #[DataProvider('provideSerializableData')]
     public function testUnserializeSerializedData($data): void
     {
         $serializer = $this->createSerializer();
@@ -48,8 +47,8 @@
 
     /**
      * @depends testInvalidDataToUnserialize
-     * @dataProvider provideSerializableData
      */
+    #[DataProvider('provideSerializableData')]
     public function testValidDataAfterInvalidData($data): void
     {
         $serializer = $this->createSerializer(self::THROW_ON_ERROR, self::THROW_ON_ERROR);
