--- ./libinput-gestures	2016-11-25 11:23:59.000000000 +0200
+++ ./libinput-gestures	2016-12-09 00:24:35.346709589 +0200
@@ -48,9 +48,8 @@
     result.stderr = None
     result.returncode = 0
     try:
-        result.stdout = subprocess.check_output(*args, **kwargs,
-                stderr=(None if check else subprocess.DEVNULL),
-                universal_newlines=True)
+        result.stdout = subprocess.check_output(stderr=(None if check else subprocess.DEVNULL),
+                universal_newlines=True, *args, **kwargs)
     except Exception as e:
         result.stderr = str(e)
         result.returncode = 1
