--- ../src/vendor/autoload.php	2024-07-12 02:14:29.615230432 +0300
+++ ../src/vendor/autoload.php	2024-07-12 02:14:20.237149761 +0300
@@ -890,3 +890,26 @@
     __DIR__
 );
 // @codeCoverageIgnoreEnd
+
+// Dependencies
+\Fedora\Autoloader\Dependencies::required([
+    __DIR__ . '/../../SebastianBergmann/CodeCoverage/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/FileIterator/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Template/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Timer/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/CliParser/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/CodeUnit/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Invoker/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Diff/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Comparator/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Environment/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Exporter/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/GlobalState/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/ObjectEnumerator/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/RecursionContext/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Type/autoload.php',
+    __DIR__ . '/../../SebastianBergmann/Version/autoload.php',
+    __DIR__ . '/../../myclabs/DeepCopy/autoload.php',
+    __DIR__ . '/../../PharIo/Manifest/autoload.php',
+    __DIR__ . '/../../PharIo/Version/autoload.php',
+]);
