--- host-linux/chrome-token-signing.pro.org	2015-10-11 00:13:52.356077985 +0300
+++ host-linux/chrome-token-signing.pro	2015-10-11 00:14:42.850075734 +0300
@@ -18,4 +18,6 @@
 hostconf.files += ee.ria.esteid.json
 extension.path = /opt/google/chrome/extensions
 extension.files += ../ckjefchnfjhjfedoccjbhjpbncimppeg.json
-INSTALLS += target hostconf
+hostconf2.path = /etc/chromium/native-messaging-hosts
+hostconf2.files += ee.ria.esteid.json
+INSTALLS += target hostconf hostconf2
