--- ../mighty-scroll.tmux	2024-12-26 19:50:22.602132154 +0200
+++ ../mighty-scroll.tmux	2024-12-26 19:51:09.158604357 +0200
@@ -8,18 +8,7 @@
     set_tmux_environment "PSCHECK" "$CURRENT_DIR/pscheck.sh"
     ;;
   *)
-    CC="${CC:-cc}"
-    if which "$CC" >/dev/null 2>&1; then
-      SRC="$CURRENT_DIR/pscheck.c"
-      OUT="$CURRENT_DIR/pscheck"
-      if [ ! -e "$OUT" ] || [ "$SRC" -nt "$OUT" ]; then
-        CFLAGS="-O3 $CFLAGS"
-        "$CC" -Wall -Wextra -Werror -Wconversion -pedantic -std=c99 $CFLAGS "$SRC" -o "$OUT" 2>&1
-      fi
-      set_tmux_environment "PSCHECK" "$OUT"
-    else
-      set_tmux_environment "PSCHECK" "$CURRENT_DIR/pscheck.sh"
-    fi
+    set_tmux_environment "PSCHECK" "/usr/bin/pscheck"
     ;;
 esac
 
