diff -r e48278058360 src/Makefile
--- a/src/Makefile	Fri Sep 06 08:59:41 2013 +0300
+++ b/src/Makefile	Fri Sep 06 09:00:59 2013 +0300
@@ -4,7 +4,7 @@
 STATIC_OBJ := $(SRC:.cpp=.o)
 SHARED_OBJ := $(SRC:.cpp=.lo)
 
-override CPPFLAGS += -I../include -DNDEBUG
+override CPPFLAGS += -I../include -DDEBUG -g3
 
 STATIC_LIB := libloki.a
 
diff -r e48278058360 test/Makefile.common
--- a/test/Makefile.common	Fri Sep 06 08:59:41 2013 +0300
+++ b/test/Makefile.common	Fri Sep 06 09:00:59 2013 +0300
@@ -1,6 +1,6 @@
 include ../../Makefile.common
 
-override CPPFLAGS += -I../../include -DNDEBUG
+override CPPFLAGS += -I../../include -DDEBUG -g3
 LDLIBS := ../../lib/libloki.a
 
 ifeq ($(OS),Windows)
