diff -r c18a3402784c Makefile.common
--- a/Makefile.common	Fri Sep 06 08:37:09 2013 +0300
+++ b/Makefile.common	Fri Sep 06 08:41:27 2013 +0300
@@ -2,6 +2,6 @@
 OS ?= $(shell uname -s)
 
 CXXWARNFLAGS := -Wall -Wold-style-cast -Wundef -Wsign-compare -Wconversion -Wpointer-arith -pedantic
-CXXFLAGS := $(CXXWARNFLAGS) -g -O2
+CXXFLAGS += $(CXXWARNFLAGS)
 
 prefix := /usr
diff -r c18a3402784c include/Makefile
--- a/include/Makefile	Fri Sep 06 08:37:09 2013 +0300
+++ b/include/Makefile	Fri Sep 06 08:41:27 2013 +0300
@@ -2,10 +2,10 @@
 
 .PHONY: install
 install:
-	mkdir -p $(prefix)/include/loki
-	mkdir -p $(prefix)/include/loki/flex
-	mkdir -p $(prefix)/include/loki/yasli
+	mkdir -p $(DESTDIR)$(prefix)/include/loki
+	mkdir -p $(DESTDIR)$(prefix)/include/loki/flex
+	mkdir -p $(DESTDIR)$(prefix)/include/loki/yasli
 	
-	install -m 644 loki/*.h $(prefix)/include/loki
-	install -m 644 loki/flex/*.h $(prefix)/include/loki/flex
-	install -m 644 loki/yasli/*.h $(prefix)/include/loki/yasli
+	install -m 644 loki/*.h $(DESTDIR)$(prefix)/include/loki
+	install -m 644 loki/flex/*.h $(DESTDIR)$(prefix)/include/loki/flex
+	install -m 644 loki/yasli/*.h $(DESTDIR)$(prefix)/include/loki/yasli
diff -r c18a3402784c src/Makefile
--- a/src/Makefile	Fri Sep 06 08:37:09 2013 +0300
+++ b/src/Makefile	Fri Sep 06 08:41:27 2013 +0300
@@ -74,13 +74,13 @@
 install: install-static install-shared
 
 install-static: $(RESULT_DIR)$(STATIC_LIB)
-	mkdir -p $(prefix)/lib
-	$(INSTALL_DATA) $(RESULT_DIR)$(STATIC_LIB) $(prefix)/lib
+	mkdir -p $(DESTDIR)$(prefix)/lib
+	$(INSTALL_DATA) $(RESULT_DIR)$(STATIC_LIB) $(DESTDIR)$(prefix)/lib
 
 install-shared: $(RESULT_DIR)$(SHARED_LIB_VERSIONED)
-	mkdir -p $(prefix)/lib
-	$(INSTALL_DATA) $(RESULT_DIR)$(SHARED_LIB_VERSIONED) $(prefix)/lib
-	cd $(prefix)/lib; ln -s $(SHARED_LIB_VERSIONED) $(SHARED_LIB_BASE)
+	mkdir -p $(DESTDIR)$(prefix)/lib
+	$(INSTALL_DATA) $(RESULT_DIR)$(SHARED_LIB_VERSIONED) $(DESTDIR)$(prefix)/lib
+	cd $(DESTDIR)$(prefix)/lib; ln -s $(SHARED_LIB_VERSIONED) $(SHARED_LIB_BASE)
 
 %.lo : %.cpp
 	$(CXX) -c $(CXXFLAGS) -fPIC $(CPPFLAGS) -o $@ $<
diff -r c18a3402784c test/Longevity/Makefile
--- a/test/Longevity/Makefile	Fri Sep 06 08:37:09 2013 +0300
+++ b/test/Longevity/Makefile	Fri Sep 06 08:41:27 2013 +0300
@@ -6,7 +6,6 @@
 BIN2 := main2$(BIN_SUFFIX)
 SRC2 := main2.cpp
 OBJ2 := $(SRC2:.cpp=.o)
-CXXFLAGS := $(CXXWARNFLAGS) -g -fexpensive-optimizations -O3
 
 .PHONY: all clean
 all: $(BIN1) $(BIN2)
diff -r c18a3402784c test/SafeFormat/Makefile
--- a/test/SafeFormat/Makefile	Fri Sep 06 08:37:09 2013 +0300
+++ b/test/SafeFormat/Makefile	Fri Sep 06 08:41:27 2013 +0300
@@ -3,7 +3,6 @@
 BIN := main$(BIN_SUFFIX)
 SRC := main.cpp
 OBJ := $(SRC:.cpp=.o)
-CXXFLAGS := $(CXXWARNFLAGS) -g -fexpensive-optimizations -O3
 
 .PHONY: all clean
 all: $(BIN)
diff -r c18a3402784c test/SmallObj/Makefile
--- a/test/SmallObj/Makefile	Fri Sep 06 08:37:09 2013 +0300
+++ b/test/SmallObj/Makefile	Fri Sep 06 08:41:27 2013 +0300
@@ -6,7 +6,6 @@
 BIN2 := SmallObjSingleton$(BIN_SUFFIX)
 SRC2 := SmallObjSingleton.cpp
 OBJ2 := $(SRC2:.cpp=.o)
-CXXFLAGS := $(CXXWARNFLAGS) -g -fexpensive-optimizations -O3
 
 .PHONY: all clean
 all: $(BIN1) $(BIN2)
