--- ../tests/unit/PseudoTypes/NonEmptyListTest.php	2026-07-04 16:05:01.398730510 +0300
+++ ../tests/unit/PseudoTypes/NonEmptyListTest.php	2026-07-04 16:38:19.573565504 +0300
@@ -19,6 +19,7 @@
 use phpDocumentor\Reflection\Types\Mixed_;
 use phpDocumentor\Reflection\Types\String_;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 final class NonEmptyListTest extends TestCase
 {
@@ -33,9 +34,7 @@
         $this->assertEquals(new Array_(null, new Integer()), $type->underlyingType());
     }
 
-    /**
-     * @dataProvider provideToStringData
-     */
+    #[DataProvider('provideToStringData')]
     public function testToString(NonEmptyList $array, string $expectedString): void
     {
         $this->assertSame($expectedString, (string) $array);
@@ -44,7 +43,7 @@
     /**
      * @return array<string, array{NonEmptyList, string}>
      */
-    public function provideToStringData(): array
+    public static function provideToStringData(): array
     {
         return [
             'simple non-empty-list' => [new NonEmptyList(), 'non-empty-list'],
