--- ../tests/unit/PseudoTypes/EnumStringTest.php	2026-07-04 16:21:24.624467508 +0300
+++ ../tests/unit/PseudoTypes/EnumStringTest.php	2026-07-04 16:21:56.981064690 +0300
@@ -18,6 +18,7 @@
 use phpDocumentor\Reflection\Types\Object_;
 use phpDocumentor\Reflection\Types\String_;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 final class EnumStringTest extends TestCase
 {
@@ -30,9 +31,7 @@
         $this->assertEquals(new String_(), $type->underlyingType());
     }
 
-    /**
-     * @dataProvider provideToStringData
-     */
+    #[DataProvider('provideToStringData')]
     public function testToString(EnumString $type, string $expectedString): void
     {
         $this->assertSame($expectedString, (string) $type);
@@ -41,7 +40,7 @@
     /**
      * @return array<string, array{EnumString, string}>
      */
-    public function provideToStringData(): array
+    public static function provideToStringData(): array
     {
         return [
             'basic' => [new EnumString(), 'enum-string'],
