--- /dev/null	2024-07-03 22:45:29.786539450 +0300
+++ ../test/conftest.py	2024-07-06 19:46:47.860247484 +0300
@@ -0,0 +1,17 @@
+import pytest
+
+@pytest.fixture(autouse=True)
+def test_verbosity_level(request):
+    """Report the verbosity level for a given pytest run.
+
+    For example:
+
+    $ pytest -vv
+    Has a verbosity level of 2
+
+    While:
+
+    $ pytest
+    Has a verbosity level of 0
+    """
+    return request.config.getoption("verbose")
