diff --git a/tests/test_description.py b/tests/test_description.py
index 9e7e4dc322..3b7d95e0a1 100644
--- a/tests/test_description.py
+++ b/tests/test_description.py
@@ -26,7 +26,7 @@
 ``` json
 [{
     "alpha": 1,
-    "beta: "this is a string"
+    "beta": "this is a string"
 }]
 ```"""
 
@@ -48,20 +48,18 @@
 <div class="highlight"><pre><span></span><span class="p">[{</span><br />\
     <span class="nt">&quot;alpha&quot;</span><span class="p">:</span>\
  <span class="mi">1</span><span class="p">,</span><br />\
-    <span class="nt">&quot;beta: &quot;</span><span class="err">this\
-</span> <span class="err">is</span> <span class="err">a</span> \
-<span class="err">string&quot;</span><br /><span class="p">}]</span>\
-<br /></pre></div>
+    <span class="nt">&quot;beta&quot;</span><span class="p">:</span>\
+ <span class="s2">&quot;this is a string&quot;</span><br />\
+<span class="p">}]</span><br /></pre></div>
 <p><br /></p>"""
 
 MARKDOWN_lt_33 = """
 <div class="highlight"><pre><span></span><span class="p">[{</span><br />\
     <span class="nt">&quot;alpha&quot;</span><span class="p">:</span>\
  <span class="mi">1</span><span class="p">,</span><br />\
-    <span class="nt">&quot;beta: &quot;</span><span class="err">this\
-</span> <span class="err">is</span> <span class="err">a</span>\
- <span class="err">string&quot;</span><br /><span class="p">}]</span>\
-<br /></pre></div>
+    <span class="nt">&quot;beta&quot;</span><span class="p">:</span>\
+ <span class="s2">&quot;this is a string&quot;</span><br />\
+<span class="p">}]</span><br /></pre></div>
 
 <p><br /></p>"""
 
@@ -112,7 +110,7 @@ class MockView(APIView):
             ``` json
             [{
                 "alpha": 1,
-                "beta: "this is a string"
+                "beta": "this is a string"
             }]
             ```"""
 
