--- ../phpunit.xml.dist	2026-08-29 12:15:55.403024428 +0300
+++ ../phpunit.xml.dist	2026-08-29 12:16:04.336971110 +0300
@@ -26,7 +26,7 @@
     </groups>
 
     <source ignoreSuppressionOfDeprecations="true">
-        <deprecationTrigger ignoreUndefinedTriggers="true">
+        <deprecationTrigger>
             <function>trigger_deprecation</function>
             <method>Doctrine\Deprecations\Deprecation::trigger</method>
             <method>Doctrine\Deprecations\Deprecation::triggerIfCalledFromOutside</method>
