--- ../src/libmongoc/tests/import-tests.cmake	2026-07-15 18:25:32.781579421 +0300
+++ ../src/libmongoc/tests/import-tests.cmake	2026-07-15 18:26:04.943975585 +0300
@@ -13,16 +13,6 @@
 )
 
 add_test_cmake_project(
-   mongoc/CMake/bare-mongoc-import src/libmongoc/tests/cmake-import
-   INSTALL_PARENT
-   SETTINGS
-      FIND_MONGOC=1
-      "FIND_MONGOC_ARGS=REQUIRED"
-      "EXPECT_BSON_VERSION=${mongo-c-driver_VERSION_FULL}"
-      "EXPECT_MONGOC_VERSION=${mongo-c-driver_VERSION_FULL}"
-)
-
-add_test_cmake_project(
    mongoc/CMake/bson-import-${PROJECT_VERSION_MAJOR}.0 src/libmongoc/tests/cmake-import
    INSTALL_PARENT
    SETTINGS
@@ -131,21 +121,3 @@
       "EXPECT_BSON_VERSION=${mongo-c-driver_VERSION_FULL}"
       EXPECT_BSON_STATIC=1
 )
-
-add_test_cmake_project(
-   mongoc/pkg-config/mongoc-import-shared src/libmongoc/tests/pkg-config-import
-   INSTALL_PARENT
-   SETTINGS
-      PKG_CONFIG_SEARCH=mongoc${PROJECT_VERSION_MAJOR}
-      "EXPECT_MONGOC_VERSION=${mongo-c-driver_VERSION_FULL}"
-)
-
-add_test_cmake_project(
-   mongoc/pkg-config/mongoc-import-static src/libmongoc/tests/pkg-config-import
-   INSTALL_PARENT
-   SETTINGS
-      PKG_CONFIG_SEARCH=mongoc${PROJECT_VERSION_MAJOR}-static
-      "EXPECT_MONGOC_VERSION=${mongo-c-driver_VERSION_FULL}"
-      EXPECT_BSON_STATIC=1
-      EXPECT_MONGOC_STATIC=1
-)
