--- ../tests/unit/DocBlock/Tags/Factory/ParamFactoryTest.php	2026-07-05 16:49:08.287945559 +0300
+++ ../tests/unit/DocBlock/Tags/Factory/ParamFactoryTest.php	2026-07-05 16:49:46.561849254 +0300
@@ -25,6 +25,7 @@
 use phpDocumentor\Reflection\Types\Mixed_;
 use phpDocumentor\Reflection\Types\Object_;
 use phpDocumentor\Reflection\Types\String_;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 final class ParamFactoryTest extends TagFactoryTestCase
 {
@@ -32,8 +33,8 @@
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\Factory\ParamFactory::__construct
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\Factory\ParamFactory::create
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\Factory\ParamFactory::supports
-     * @dataProvider paramInputProvider
      */
+    #[DataProvider('paramInputProvider')]
     public function testParamIsCreated(string $input, Tag $expected): void
     {
         $ast = $this->parseTag($input);
@@ -50,7 +51,7 @@
     /**
      * @return array<int, array<int, string|Param|InvalidTag>>
      */
-    public function paramInputProvider(): array
+    public static function paramInputProvider(): array
     {
         return [
             [
