--- ../tests/unit/DocBlock/Tags/MethodParameterTest.php	2026-07-05 16:50:14.106998717 +0300
+++ ../tests/unit/DocBlock/Tags/MethodParameterTest.php	2026-07-05 16:51:18.069905740 +0300
@@ -24,6 +24,7 @@
 use phpDocumentor\Reflection\Types\Object_;
 use phpDocumentor\Reflection\Types\String_;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 use stdClass;
 
 use function sprintf;
@@ -43,7 +44,7 @@
     }
 
     /** @return array<array{0: Type, 1: mixed, 2: string}> */
-    public function collectionDefaultValuesProvider(): array
+    public static function collectionDefaultValuesProvider(): array
     {
         return [
             [new String_(), '1', '\'1\''],
@@ -66,10 +67,10 @@
      *
      * @param mixed $defaultValue
      *
-     * @dataProvider collectionDefaultValuesProvider
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\BaseTag::render
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\BaseTag::getName
      */
+    #[DataProvider('collectionDefaultValuesProvider')]
     public function testIfTagCanBeRenderedUsingMethodParameterWithDefaultValue(
         Type $type,
         $defaultValue,
