--- ../tests/unit/DocBlock/Tags/Factory/TemplateFactoryTest.php	2026-07-05 16:40:33.466181994 +0300
+++ ../tests/unit/DocBlock/Tags/Factory/TemplateFactoryTest.php	2026-07-05 16:41:17.201070247 +0300
@@ -20,6 +20,7 @@
 use phpDocumentor\Reflection\Types\Context;
 use phpDocumentor\Reflection\Types\Mixed_;
 use phpDocumentor\Reflection\Types\Object_;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 final class TemplateFactoryTest extends TagFactoryTestCase
 {
@@ -27,8 +28,8 @@
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\Factory\TemplateFactory::__construct
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\Factory\TemplateFactory::create
      * @covers \phpDocumentor\Reflection\DocBlock\Tags\Factory\TemplateFactory::supports
-     * @dataProvider templateInputProvider
      */
+    #[DataProvider('templateInputProvider')]
     public function testTemplateIsCreated(string $input, Tag $expected): void
     {
         $ast = $this->parseTag($input);
@@ -45,7 +46,7 @@
     /**
      * @return array<int, array<int, string|Template>>
      */
-    public function templateInputProvider(): array
+    public static function templateInputProvider(): array
     {
         return [
             [
