--- ../tests/test_recursive.py   2020-06-14 21:32:54.465032551 +0300
+++ ../tests/test_recursive.py   2020-06-14 21:33:15.096158197 +0300
@@ -41,7 +41,7 @@
 
 class RecursiveModel(models.Model):
     name = models.CharField(max_length=255)
-    parent = models.ForeignKey('self', null=True)
+    parent = models.ForeignKey('self', null=True, on_delete=models.CASCADE)
 
 
 class RecursiveModelSerializer(serializers.ModelSerializer):
