--- ../module/kvmfr.c	2024-06-30 20:08:30.905402631 +0300
+++ ../module/kvmfr.c	2024-06-30 20:09:33.618535086 +0300
@@ -539,7 +539,11 @@
   if (kvmfr->major < 0)
     goto out_free;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
   kvmfr->pClass = class_create(THIS_MODULE, KVMFR_DEV_NAME);
+#else
+  kvmfr->pClass = class_create(KVMFR_DEV_NAME);
+#endif
   if (IS_ERR(kvmfr->pClass))
     goto out_unreg;
 
