diff --git a/linphone-app/src/app/cli/Cli.cpp b/linphone-app/src/app/cli/Cli.cpp
index 3d84ad63dbb149da63de73ebdabd5cbf35b0a113..9193f3d4887a913d9be3ba5bcb4c6d130841f6d8 100644
--- a/linphone-app/src/app/cli/Cli.cpp
+++ b/linphone-app/src/app/cli/Cli.cpp
@@ -182,7 +182,7 @@ static void cliInitiateConference (QHash<QString, QString> &args) {
 
 	qInfo() << QStringLiteral("Create conference with id: `%1`.").arg(id);
 	auto confParameters = core->createConferenceParams();
-	confParameters->setVideoEnabled(false);// Video is not yet fully supported by the application in conference
+	confParameters->enableVideo(false);// Video is not yet fully supported by the application in conference
 	conference = core->createConferenceWithParams(confParameters);
 	conference->setId(Utils::appStringToCoreString(id));
 
diff --git a/linphone-app/src/components/conference/ConferenceAddModel.cpp b/linphone-app/src/components/conference/ConferenceAddModel.cpp
index 5087be272527bb2bcf4a83f392f10da1efeede9a..b67bb71341eb22be268d438e4b818edc19f7081d 100644
--- a/linphone-app/src/components/conference/ConferenceAddModel.cpp
+++ b/linphone-app/src/components/conference/ConferenceAddModel.cpp
@@ -149,7 +149,7 @@ void ConferenceHelperModel::ConferenceAddModel::update () {
   shared_ptr<linphone::Conference> conference = mConferenceHelperModel->mCore->getConference();
   if(!conference){
     auto parameters = mConferenceHelperModel->mCore->createConferenceParams();
-    parameters->setVideoEnabled(false);// Video is not yet fully supported by the application in conference
+    parameters->enableVideo(false);// Video is not yet fully supported by the application in conference
     conference = mConferenceHelperModel->mCore->createConferenceWithParams(parameters);
   }
   auto currentCalls = CoreManager::getInstance()->getCore()->getCalls();
