Description: Avoid conflicting wchar_t declarations
Author: Sébastien Villemot <sebastien@debian.org>
Bug-Debian: https://bugs.debian.org/972965
Forwarded: no

--- a/LIBRARY/src/bid_functions.h
+++ b/LIBRARY/src/bid_functions.h
@@ -36,16 +36,9 @@
 #ifndef _BID_FUNCTIONS_H
 #define _BID_FUNCTIONS_H
 
-#if !defined (__GNUC__) || defined(__QNX__)
 #include <wchar.h>
-#endif
 #include <ctype.h>
 
-// Fix system header issue on Sun solaris and define required type by ourselves
-#if !defined(_WCHAR_T) && !defined(_WCHAR_T_DEFINED) && !defined(__QNX__)
-typedef int   wchar_t;
-#endif
-
 
 #ifdef IN_LIBGCC2
 // When we are built as the part of the gcc runtime library, libgcc,
