--- ../Tests/Hasher/SodiumPasswordHasherTest.php	2026-07-07 20:55:54.953164656 +0300
+++ ../Tests/Hasher/SodiumPasswordHasherTest.php	2026-07-07 20:56:05.093665877 +0300
@@ -74,7 +74,7 @@
         $this->assertTrue($hasher->verify((new NativePasswordHasher(null, null, 4, \PASSWORD_BCRYPT))->hash($plainPassword), $plainPassword));
     }
 
-    #[RequiresPhp('<8.4')]
+    #[RequiresPhp('<8.4.0')]
     public function testBcryptWithNulByteWithNativePasswordHash()
     {
         $hasher = new SodiumPasswordHasher(null, null);
