--- ../Tests/Hasher/NativePasswordHasherTest.php	2026-07-07 20:53:51.721607094 +0300
+++ ../Tests/Hasher/NativePasswordHasherTest.php	2026-07-07 20:54:00.214349721 +0300
@@ -100,7 +100,7 @@
         $this->assertTrue($hasher->verify($hasher->hash($plainPassword), $plainPassword));
     }
 
-    #[RequiresPhp('<8.4')]
+    #[RequiresPhp('<8.4.0')]
     public function testBcryptWithNulByteWithNativePasswordHash()
     {
         $hasher = new NativePasswordHasher(null, null, 4, \PASSWORD_BCRYPT);
