--- ../tests/Mapping/DriverChainTest.php	2026-07-19 11:48:09.847229921 +0300
+++ ../tests/Mapping/DriverChainTest.php	2026-07-19 11:51:46.042917524 +0300
@@ -11,14 +11,13 @@
 use Doctrine\Tests\Persistence\Mapping\Fixtures\Manager\Manager;
 use Doctrine\Tests\Persistence\Mapping\Fixtures\Model;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\TestWith;
 use stdClass;
 
 class DriverChainTest extends TestCase
 {
-    /**
-     * @testWith ["Doctrine\\Tests\\Models\\Company", "Doctrine\\Tests\\Persistence\\Mapping"]
-     *           ["Doctrine\\Tests\\Persistence\\Map\\", "Doctrine\\Tests\\Persistence\\Map"]
-     */
+    #[TestWith(['Doctrine\\Tests\\Models\\Company', 'Doctrine\\Tests\\Persistence\\Mapping'])]
+    #[TestWith(['Doctrine\\Tests\\Persistence\\Map\\', 'Doctrine\\Tests\\Persistence\\Map'])]
     public function testDelegateToMatchingNamespaceDriver(string $namespace1, string $namespace2): void
     {
         $className     = DriverChainEntity::class;
@@ -67,12 +66,12 @@
         $driver1 = $this->createMock(MappingDriver::class);
         $driver1->expects(self::once())
                 ->method('getAllClassNames')
-                ->will(self::returnValue(['Doctrine\Tests\Models\Company\Foo']));
+                ->willReturn(['Doctrine\Tests\Models\Company\Foo']);
 
         $driver2 = $this->createMock(MappingDriver::class);
         $driver2->expects(self::once())
                 ->method('getAllClassNames')
-                ->will(self::returnValue(['Doctrine\Tests\ORM\Mapping\Bar', 'Doctrine\Tests\ORM\Mapping\Baz', 'FooBarBaz']));
+                ->willReturn(['Doctrine\Tests\ORM\Mapping\Bar', 'Doctrine\Tests\ORM\Mapping\Baz', 'FooBarBaz']);
 
         $chain->addDriver($driver1, 'Doctrine\Tests\Models\Company');
         $chain->addDriver($driver2, 'Doctrine\Tests\ORM\Mapping');
@@ -108,14 +107,14 @@
         $companyDriver->expects(self::once())
             ->method('isTransient')
             ->with(self::equalTo($managerClassName))
-            ->will(self::returnValue(false));
+            ->willReturn(false);
 
         $defaultDriver->expects(self::never())
             ->method('loadMetadataForClass');
         $defaultDriver->expects(self::once())
             ->method('isTransient')
             ->with(self::equalTo($entityClassName))
-            ->will(self::returnValue(true));
+            ->willReturn(true);
 
         self::assertNull($chain->getDefaultDriver());
 
@@ -138,11 +137,11 @@
 
         $companyDriver->expects(self::once())
             ->method('getAllClassNames')
-            ->will(self::returnValue(['Doctrine\Tests\Models\Company\Foo']));
+            ->willReturn(['Doctrine\Tests\Models\Company\Foo']);
 
         $defaultDriver->expects(self::once())
             ->method('getAllClassNames')
-            ->will(self::returnValue(['Other\Class']));
+            ->willReturn(['Other\Class']);
 
         $chain->setDefaultDriver($defaultDriver);
         $chain->addDriver($companyDriver, 'Doctrine\Tests\Models\Company');
