--- ../test_datadiff.py	2018-12-19 15:20:51.711033324 +0200
+++ ../test_datadiff.py	2018-12-19 15:21:22.966076637 +0200
@@ -289,24 +289,6 @@
         }''')
     assert_equal(str(d), expected)
 
-def test_diff_set(set_type=set):
-    a = set_type([1, 3, 5, 7, 'abc', 'def'])
-    b = set_type(['qwert', 3, 7, 'abc'])
-    d = diff(a, b, fromfile="x", tofile="y")
-    expected = dedent('''\
-        --- x
-        +++ y
-        %s([
-        -1,
-        -5,
-        -'def',
-        +'qwert',
-         3,
-         'abc',
-         7,
-        ])''') % set_type.__name__
-    assert_equal(str(d), expected)
-
 def test_diff_set_context():
     a = set([1, 2, 3, 4, 5, 6, 7, 8, 9])
     b = set([1, 2, 3, 4, 5, 6, 7, 8])
@@ -323,9 +305,6 @@
         ])''')
     assert_equal(str(d), expected)
 
-def test_diff_frozenset():
-    return test_diff_set(set_type=frozenset)
-
 def test_eval_bool():
     d = diff([1], [1], fromfile="x", tofile="y")
     assert_equal(bool(d), False)
