--- ../tests/MultipartStreamTest.php	2026-07-14 00:04:27.312129131 +0300
+++ ../tests/MultipartStreamTest.php	2026-07-14 00:05:10.694602586 +0300
@@ -7,6 +7,7 @@
 use GuzzleHttp\Psr7;
 use GuzzleHttp\Psr7\MultipartStream;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 class MultipartStreamTest extends TestCase
 {
@@ -655,9 +656,7 @@
         self::assertSame($expected, (string) $b);
     }
 
-    /**
-     * @dataProvider unstringableCustomHeaderValueProvider
-     */
+    #[DataProvider('unstringableCustomHeaderValueProvider')]
     public function testRejectsUnstringableCustomHeaderValues($value): void
     {
         $this->expectException(\InvalidArgumentException::class);
