--- ../tests/UtilsTest.php	2026-07-13 21:46:58.410027847 +0300
+++ ../tests/UtilsTest.php	2026-07-13 21:48:24.246976372 +0300
@@ -8,6 +8,7 @@
 use GuzzleHttp\Psr7\FnStream;
 use GuzzleHttp\Psr7\NoSeekStream;
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 use Psr\Http\Message\StreamInterface;
 use Psr\Http\Message\UriInterface;
 
@@ -733,9 +734,7 @@
         self::assertSame('1', $modified->getHeaderLine('X-Test'));
     }
 
-    /**
-     * @dataProvider hostHeaderCaseProvider
-     */
+    #[DataProvider('hostHeaderCaseProvider')]
     public function testModifyRequestRejectsUriAndExplicitHostHeader(string $hostHeader): void
     {
         $request = new Psr7\Request('GET', 'http://foo.com');
@@ -942,10 +941,9 @@
     }
 
     /**
-     * @dataProvider providesCaselessRemoveCases
-     *
      * @param string[] $keys
      */
+    #[DataProvider('providesCaselessRemoveCases')]
     public function testCaselessRemove(array $keys, array $data, array $expected): void
     {
         self::assertSame($expected, Psr7\Utils::caselessRemove($keys, $data));
