--- ../phpunit.xml.dist	2026-08-29 01:16:13.325422367 +0300
+++ ../phpunit.xml.dist	2026-08-29 01:16:21.119499543 +0300
@@ -20,7 +20,7 @@
     </testsuites>
 
     <source ignoreSuppressionOfDeprecations="true">
-        <deprecationTrigger ignoreUndefinedTriggers="true">
+        <deprecationTrigger>
             <function>trigger_deprecation</function>
             <method>Doctrine\Deprecations\Deprecation::trigger</method>
             <method>Doctrine\Deprecations\Deprecation::triggerIfCalledFromOutside</method>
