--- ../tests/ManifestDocumentMapperTest.php	2026-08-09 15:36:05.039196503 +0300
+++ ../tests/ManifestDocumentMapperTest.php	2026-08-09 15:36:46.025413159 +0300
@@ -10,6 +10,8 @@
  */
 namespace PharIo\Manifest;
 
+use PHPUnit\Framework\Attributes\DataProvider;
+
 /**
  * @covers \PharIo\Manifest\ManifestDocumentMapper
  *
@@ -52,14 +54,13 @@
  */
 class ManifestDocumentMapperTest extends \PHPUnit\Framework\TestCase {
     /**
-     * @dataProvider dataProvider
-     *
      * @uses         \PharIo\Manifest\Application
      * @uses         \PharIo\Manifest\ApplicationName
      * @uses         \PharIo\Manifest\Library
      * @uses         \PharIo\Manifest\Extension
      * @uses         \PharIo\Manifest\ExtensionElement
      */
+    #[DataProvider('dataProvider')]
     public function testCanSerializeToString($expected): void {
         $manifestDocument = ManifestDocument::fromFile($expected);
         $mapper           = new ManifestDocumentMapper();
@@ -70,7 +71,7 @@
         );
     }
 
-    public function dataProvider() {
+    public static function dataProvider() {
         return [
             'application'   => [__DIR__ . '/_fixture/phpunit-5.6.5.xml'],
             'library'       => [__DIR__ . '/_fixture/library.xml'],
