--- ../test/HttpResponseTest.php	2026-07-11 10:47:57.896754627 +0300
+++ ../test/HttpResponseTest.php	2026-07-11 10:48:38.312273360 +0300
@@ -3,6 +3,7 @@
 namespace Amp\Http;
 
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Attributes\DataProvider;
 
 class TestHttpResponse extends HttpResponse
 {
@@ -38,7 +39,7 @@
         self::assertSame($reason, $response->getReason());
     }
 
-    public function provideStatuses(): iterable
+    public static function provideStatuses(): iterable
     {
         yield 'informational' => [100, true, false, false, false, false];
         yield 'success' => [200, false, true, false, false, false];
@@ -47,9 +48,7 @@
         yield 'server-error' => [500, false, false, false, false, true];
     }
 
-    /**
-     * @dataProvider provideStatuses
-     */
+    #[DataProvider('provideStatuses')]
     public function testStatusMethods(
         int $code,
         bool $info,
